Index: markup.ml-0.8.1/test/dune
===================================================================
--- markup.ml-0.8.1.orig/test/dune
+++ markup.ml-0.8.1/test/dune
@@ -1,6 +1,6 @@
 (executable
  (name test)
- (libraries markup oUnit test_support)
+ (libraries markup ounit2 test_support)
  (flags (:standard -w +A-44-48)))
 
 (alias
Index: markup.ml-0.8.1/test/lwt_unix/dune
===================================================================
--- markup.ml-0.8.1.orig/test/lwt_unix/dune
+++ markup.ml-0.8.1/test/lwt_unix/dune
@@ -1,6 +1,6 @@
 (executable
  (name test_lwt)
- (libraries lwt.unix markup-lwt.unix oUnit test_support)
+ (libraries lwt.unix markup-lwt.unix ounit2 test_support)
  (flags (:standard -w +A-44)))
 
 (alias
Index: markup.ml-0.8.1/test/support/dune
===================================================================
--- markup.ml-0.8.1.orig/test/support/dune
+++ markup.ml-0.8.1/test/support/dune
@@ -1,4 +1,4 @@
 (library
  (name test_support)
- (libraries markup oUnit)
+ (libraries markup ounit2)
  (flags (:standard -w +A-44)))
