Index: markup.ml-0.7.5/_tags
===================================================================
--- markup.ml-0.7.5.orig/_tags
+++ markup.ml-0.7.5/_tags
@@ -3,13 +3,13 @@
 
 <src/*> or <test/**>: package(uutf)
 <src/markup_lwt*>: package(lwt)
-<src/markup_lwt_unix*>: package(lwt.unix)
+<src/markup_lwt_unix*>: thread, package(lwt.unix)
 
 <src/translate_entities.*>: package(yojson)
 
 <test/*>: package(oUnit)
-<test/test_lwt.*>: package(lwt), package(lwt.unix)
-<test/test_asynchronous.*>: package(lwt), package(lwt.unix)
+<test/test_lwt.*>: thread, package(lwt), package(lwt.unix)
+<test/test_asynchronous.*>: thread, package(lwt), package(lwt.unix)
 <test/performance/*>: package(unix)
 <test/performance/performance_nethtml.*>: package(netstring)
 <test/performance/performance_xmlm.*>: package(xmlm)
