Index: ca-certs-0.2.3/lib/ca_certs.ml
===================================================================
--- ca-certs-0.2.3.orig/lib/ca_certs.ml
+++ ca-certs-0.2.3/lib/ca_certs.ml
@@ -75,8 +75,8 @@ let rec map_m f l =
     reencoded as a single PEM certificate. *)
 let windows_trust_anchors () =
   let* anchors = get_anchors () in
-  let* cert_list = map_m X509.Certificate.decode_der anchors in
-  Ok (X509.Certificate.encode_pem_multiple cert_list |> Cstruct.to_string)
+  let* cert_list = map_m X509.Certificate.decode_der (List.map Cstruct.to_string anchors) in
+  Ok (X509.Certificate.encode_pem_multiple cert_list)
 
 let trust_anchors () =
   if Sys.win32 then windows_trust_anchors ()
