Index: tls-0.17.5/lwt/x509_lwt.ml
===================================================================
--- tls-0.17.5.orig/lwt/x509_lwt.ml
+++ tls-0.17.5/lwt/x509_lwt.ml
@@ -72,7 +72,7 @@ let certs_of_pem_dir path =
 let crl_of_pem path =
   catch_invalid_arg
     (read_file path >|= fun data ->
-     match X509.CRL.decode_der data with
+     match X509.CRL.decode_der (Cstruct.to_string data) with
      | Ok cs -> cs
      | Error (`Msg m) -> invalid_arg ("failed to parse CRL " ^ m))
     (o failure @@ Printf.sprintf "CRL in %s: %s" path)
