Index: ocaml-conduit-6.2.3/src/conduit-lwt-unix/conduit_lwt_tls.real.ml
===================================================================
--- ocaml-conduit-6.2.3.orig/src/conduit-lwt-unix/conduit_lwt_tls.real.ml
+++ ocaml-conduit-6.2.3/src/conduit-lwt-unix/conduit_lwt_tls.real.ml
@@ -36,7 +36,7 @@ module Client = struct
         | None -> Lwt.return_unit
         | Some src_sa -> Lwt_unix.bind fd src_sa)
         >>= fun () ->
-        let config = Tls.Config.client ~authenticator ?certificates () in
+        let config = Result.get_ok (Tls.Config.client ~authenticator ?certificates ()) in
         Lwt_unix.connect fd sa >>= fun () ->
         Tls_lwt.Unix.client_of_fd config ~host fd >|= fun t ->
         let ic, oc = Tls_lwt.of_t t in
@@ -59,7 +59,7 @@ module Server = struct
   let init ?backlog ~certfile ~keyfile ?stop ?timeout sa callback =
     X509_lwt.private_of_pems ~cert:certfile ~priv_key:keyfile
     >>= fun certificate ->
-    let config = Tls.Config.server ~certificates:(`Single certificate) () in
+    let config = Result.get_ok (Tls.Config.server ~certificates:(`Single certificate) ()) in
     init' ?backlog ?stop ?timeout config sa callback
 end
 
