Index: atd-2.2.1/atd/src/dune
===================================================================
--- atd-2.2.1.orig/atd/src/dune
+++ atd-2.2.1/atd/src/dune
@@ -1,5 +1,5 @@
 (ocamllex lexer doc_lexer)
-(menhir (modules parser))
+(menhir (infer true) (modules parser))
 
 (library
  (name atd)
Index: atd-2.2.1/dune-project
===================================================================
--- atd-2.2.1.orig/dune-project
+++ atd-2.2.1/dune-project
@@ -1,5 +1,5 @@
 (lang dune 2.0)
-(using menhir 1.0)
+(using menhir 2.0)
 
 (name atd)
 (version 2.2.1)
