Index: opam-2.1.0/src/solver/opamCudf.ml
===================================================================
--- opam-2.1.0.orig/src/solver/opamCudf.ml
+++ opam-2.1.0/src/solver/opamCudf.ml
@@ -1323,7 +1323,7 @@ let call_external_solver ~version_map un
         | OpamCudfSolver.Timeout None -> raise (Timeout None)
       in
       let r =
-        Dose_algo.Depsolver.check_request_using ~call_solver ~criteria ~explain req
+        Dose_algo.Depsolver.check_request_using ~call_solver ~explain req
       in
       if !timed_out then raise (Timeout (Some r)) else r
     in
