Index: ppx_custom_printf-0.14.0/src/ppx_custom_printf.ml
===================================================================
--- ppx_custom_printf-0.14.0.orig/src/ppx_custom_printf.ml
+++ ppx_custom_printf-0.14.0/src/ppx_custom_printf.ml
@@ -264,7 +264,7 @@ let expand_format_string ~loc fmt_string
 let expand e =
   match e.pexp_desc with
   | Pexp_apply ({ pexp_attributes = ident_attrs; _ },
-                [ (Nolabel, { pexp_desc = Pexp_constant (Pconst_string (str, _))
+                [ (Nolabel, { pexp_desc = Pexp_constant (Pconst_string (str, _, _))
                             ; pexp_loc = loc; pexp_loc_stack = _
                             ; pexp_attributes = str_attrs }) ]) ->
     assert_no_attributes ident_attrs;
