Index: xmldiff-0.5.0/checkocaml.ml
===================================================================
--- xmldiff-0.5.0.orig/checkocaml.ml
+++ xmldiff-0.5.0/checkocaml.ml
@@ -435,7 +435,7 @@ let dummy_version = [0]
    [s] must b in the form [X[.Y[.Z[...]]]]. If the string is not correct,
    then the dummy version is returned. *)
 let version_of_string s =
-  let l = Str.split (Str.regexp_string ".") s in
+  let l = Str.split (Str.regexp_string ".") (Str.replace_first (Str.regexp_string "v") "" s) in
   try List.map int_of_string l
   with Failure _ -> dummy_version
 
