Index: camlimages-5.0.1/graphics/jbuild
===================================================================
--- camlimages-5.0.1.orig/graphics/jbuild
+++ camlimages-5.0.1/graphics/jbuild
@@ -13,7 +13,7 @@ let jbuild = {|(jbuild_version 1)
   (public_name camlimages.graphics)
   (optional)
   (wrapped false)
-  (libraries ( camlimages.core ))
+  (libraries ( camlimages.core graphics ))
   (virtual_deps (graphics))
   ))
 |}
