Index: ocaml-gettext-0.3.8/configure.in
===================================================================
--- ocaml-gettext-0.3.8.orig/configure.in
+++ ocaml-gettext-0.3.8/configure.in
@@ -109,7 +109,7 @@ OCAMLFIND_CHECK_MODULE(camomile,[],
 ])
 
 if test "x$BUILD_TEST" = "xyes"; then
-  OCAMLFIND_CHECK_MODULE(oUnit,[],[AC_MSG_ERROR(Cannot build test.)])
+  OCAMLFIND_CHECK_MODULE(ounit2,[],[AC_MSG_ERROR(Cannot build test.)])
 fi
 
 if test "x$BUILD_BENCH" = "xyes"; then
Index: ocaml-gettext-0.3.8/test/Makefile
===================================================================
--- ocaml-gettext-0.3.8.orig/test/Makefile
+++ ocaml-gettext-0.3.8/test/Makefile
@@ -31,7 +31,7 @@ else
 ifeq ($(BUILD_TEST),yes)
 NAME       = test
 CMO       += test.cmo
-REQUIRES  += oUnit
+REQUIRES  += ounit2
 endif
 endif
 
