Index: onanomsg-1.0/_tags
===================================================================
--- onanomsg-1.0.orig/_tags
+++ onanomsg-1.0/_tags
@@ -11,13 +11,13 @@ true: debug, bin_annot, safe_string
 		package(containers)
 
 
-<lib/nanomsg_lwt.*>: thread, package(lwt.unix), package(lwt.ppx)
+<lib/nanomsg_lwt.*>: thread, package(lwt.unix), package(lwt_ppx)
 
 <lib_test/*>: package(bigstring), \
 	      package(ctypes.foreign), \
 	      package(ipaddr), \
 	      package(lwt.unix), \
-	      package(lwt.ppx), \
+	      package(lwt_ppx), \
 	      package(oUnit), \
 	      package(ppx_deriving), \
 	      package(containers)
Index: onanomsg-1.0/lib_test/suite.ml
===================================================================
--- onanomsg-1.0.orig/lib_test/suite.ml
+++ onanomsg-1.0/lib_test/suite.ml
@@ -129,12 +129,12 @@ let pair_test ctx =
   let _ = connect peer2 addr in
   let rec inner () =
     Lwt_list.iter_s (fun msg ->
-        NB.send_string peer1 msg >>
+        NB.send_string peer1 msg >>= fun () ->
         let%lwt recv_msg = NB.recv_string peer2 in
         assert_equal msg recv_msg; Lwt.return_unit
-      ) msgs >>
+      ) msgs >>= fun () ->
     Lwt_list.iter_s (fun msg ->
-        NB.send_string peer2 msg >>
+        NB.send_string peer2 msg >>= fun () ->
         let%lwt recv_msg = NB.recv_string peer1 in
         assert_equal msg recv_msg; Lwt.return_unit
       ) msgs >|= fun () ->
@@ -236,7 +236,7 @@ let pipeline_local_test ctx =
   let sender addr =
     let s = socket Push in
     let _ = connect s addr in
-    Lwt_list.iter_s (Nanomsg_lwt.send_string s) @@ Array.to_list msgs >>
+    Lwt_list.iter_s (Nanomsg_lwt.send_string s) @@ Array.to_list msgs >>= fun () ->
     Lwt_unix.yield () >|= fun () -> close s
   in
   Lwt_main.run (sender (`Inproc "rdvpoint") <&> receiver (`Inproc "rdvpoint"))
