Index: calendar-2.04/configure.in
===================================================================
--- calendar-2.04.orig/configure.in
+++ calendar-2.04/configure.in
@@ -176,7 +176,7 @@ AC_MSG_RESULT(${LIBEXT})
 
 # Checking workable native dynlink
 HAS_NATDYNLINK=no
-AC_CHECK_FILE(${OCAMLLIB}/dynlink.cmxa,HAS_NATDYNLINK=yes,HAS_NATDYNLINK=no)
+AC_CHECK_FILE(${OCAMLLIB}/dynlink/dynlink.cmxa,HAS_NATDYNLINK=yes,HAS_NATDYNLINK=no)
 
 if test "$HAS_NATDYNLINK" != "no" ; then
   echo "let f x y = Dynlink.loadfile \"foo\"; ignore (Dynlink.is_native); abs_float (x -. y)" > test_dynlink.ml
