Index: ppx_deriving_yojson-3.5.1/src_test/dune
===================================================================
--- ppx_deriving_yojson-3.5.1.orig/src_test/dune
+++ ppx_deriving_yojson-3.5.1/src_test/dune
@@ -5,11 +5,11 @@
 
 (executable
  (name test_ppx_yojson)
- (libraries oUnit result)
+ (libraries ounit2 result)
  (preprocess (pps ppx_deriving.show ppx_deriving_yojson))
  (flags (:standard -w -9-39-27-34-37)))
 
 (alias
  (name runtest)
  (deps test_ppx_yojson.exe)
- (action (run %{deps})))
\ No newline at end of file
+ (action (run %{deps})))
