Index: ppx_deriving_yojson-3.6.1/src/ppx_deriving_yojson.ml
===================================================================
--- ppx_deriving_yojson-3.6.1.orig/src/ppx_deriving_yojson.ml
+++ ppx_deriving_yojson-3.6.1/src/ppx_deriving_yojson.ml
@@ -452,7 +452,7 @@ let ser_str_of_type_ext ~options ~path:_
           (* nothing to do, since the constructor must be handled in original
              constructor declaration *)
           acc_cases
-        | Pext_decl (pext_args, _) ->
+        | Pext_decl (_, pext_args, _) ->
           let json_name = attr_name name' pext_attributes in
           let case =
             match pext_args with
@@ -675,7 +675,7 @@ let desu_str_of_type_ext ~options ~path
           (* nothing to do since it must have been handled in the original
              constructor declaration *)
           acc_cases
-        | Pext_decl (pext_args, _) ->
+        | Pext_decl (_, pext_args, _) ->
           let case =
             match pext_args with
             | Pcstr_tuple(args) ->
