Index: ocaml-migrate-parsetree-2.4.0/src/config/gen.ml
===================================================================
--- ocaml-migrate-parsetree-2.4.0.orig/src/config/gen.ml
+++ ocaml-migrate-parsetree-2.4.0/src/config/gen.ml
@@ -26,6 +26,7 @@ let () =
      | (5, 0)  -> "500"
      | (5, 1)  -> "510"
      | (5, 2)  -> "520"
+     | (5, 3)  -> "530"
      | _ ->
        Printf.eprintf "Unknown OCaml version %s\n" ocaml_version_str;
        exit 1)
