Index: ppx_import-1.6.2/src/ppx_import.ml
===================================================================
--- ppx_import-1.6.2.orig/src/ppx_import.ml
+++ ppx_import-1.6.2/src/ppx_import.ml
@@ -42,7 +42,7 @@ let lazy_env = lazy (
      modules. On the other hand, setting recursive_types more often
      than necessary does not seem harmful. *)
   Clflags.recursive_types := true;
-  Compmisc.init_path false;
+  Compmisc.init_path ();
   Compmisc.initial_env ()
 )
 
