Index: a/src/gen_js_api.ml
===================================================================
--- a/src/gen_js_api.ml
+++ b/src/gen_js_api.ml
@@ -275,7 +275,7 @@ type classdecl =
   | Constructor of { class_name: string; js_class_name: string; class_arrow: arrow_params }
 
 type decl =
-  | Module of string * attributes * decl list
+  | Module of string option * attributes * decl list
   | Type of rec_flag * Parsetree.type_declaration list
   | Val of string * typ * valdef * Location.t
   | Class of classdecl list
