Index: gen_js_api-1.0.6/ppx-driver/gen_js_api_ppx_driver.ml
===================================================================
--- gen_js_api-1.0.6.orig/ppx-driver/gen_js_api_ppx_driver.ml
+++ gen_js_api-1.0.6/ppx-driver/gen_js_api_ppx_driver.ml
@@ -1,11 +1,10 @@
-module Selected = Ppxlib.Select_ast(Ppxlib_ast__Versions.OCaml_412)
+module Selected = Ppxlib.Selected_ast
 
 let copy_attribute (a : Migrate_parsetree.Ast_412.Parsetree.attribute)
   : Ppxlib.Ast.attribute =
-  let pat : Migrate_parsetree.Ast_412.Parsetree.pattern =
-    Migrate_parsetree.Migrate_411_412.copy_pattern (Ppxlib_ast.Ast_helper.Pat.any ~attrs:[Migrate_parsetree.Migrate_412_411.copy_attribute a] ())
+  let pat : Ppxlib.Parsetree.pattern =
+    (Ppxlib_ast.Ast_helper.Pat.any ~attrs:[a] ())
   in
-  let pat = Selected.Of_ocaml.copy_pattern pat in
   List.hd pat.ppat_attributes
 
 let check_attributes_with_ppxlib = false
@@ -30,6 +29,6 @@ let () =
   in
   Ppxlib.Driver.register_transformation
     "gen_js_api"
-    ~impl:(fun str -> List.map Migrate_parsetree.Migrate_412_411.copy_structure_item (mapper.structure mapper (List.map Migrate_parsetree.Migrate_411_412.copy_structure_item str)))
+    ~impl:(fun str -> (mapper.structure mapper str))
     ~intf:(fun sig_ ->
-      List.map Migrate_parsetree.Migrate_412_411.copy_signature_item (mapper_for_sig.signature mapper_for_sig (List.map Migrate_parsetree.Migrate_411_412.copy_signature_item sig_)))
+      mapper_for_sig.signature mapper_for_sig sig_)
