Index: ocamlmod-0.0.9/_tags
===================================================================
--- ocamlmod-0.0.9.orig/_tags
+++ ocamlmod-0.0.9/_tags
@@ -18,8 +18,8 @@ true: annot, bin_annot
 "src/ocamlmod.byte": pkg_str
 <src/*.ml{,i,y}>: pkg_str
 # Executable test
-"test/test.byte": pkg_oUnit
+"test/test.byte": pkg_ounit2
 "test/test.byte": pkg_str
-<test/*.ml{,i,y}>: pkg_oUnit
+<test/*.ml{,i,y}>: pkg_ounit2
 <test/*.ml{,i,y}>: pkg_str
 # OASIS_STOP
Index: ocamlmod-0.0.9/setup.ml
===================================================================
--- ocamlmod-0.0.9.orig/setup.ml
+++ ocamlmod-0.0.9/setup.ml
@@ -7216,7 +7216,7 @@ let setup_t =
                       bs_build_depends =
                         [
                            FindlibPackage
-                             ("oUnit",
+                             ("ounit2",
                                Some (OASISVersion.VGreaterEqual "2.0.0"));
                            FindlibPackage ("str", None)
                         ];
