Index: ledit-ledit-2-06/Makefile
===================================================================
--- ledit-ledit-2-06.orig/Makefile
+++ ledit-ledit-2-06/Makefile
@@ -5,9 +5,9 @@ LIBDIR=/usr/local/lib
 MANDIR=/usr/local/man/man1
 OCAMLC=ocamlc
 OCAMLOPT=ocamlopt
-CAMLP5=camlp5r -I ext -I +../camlp-streams `ocamlfind query camlp-streams`/camlp_streams.cma
+CAMLP5=camlp5r -I ext -I +camlp-streams `ocamlfind query camlp-streams`/camlp_streams.cma
 OBJS=cursor.cmo ledit.cmo go.cmo
-INCLUDES= -I `camlp5 -where` -I +unix -I +../camlp-streams
+INCLUDES= -I `camlp5 -where` -I +unix -I +camlp-streams
 OTHER_OBJS=unix.cma gramlib.cma camlp_streams.cma
 OTHER_XOBJS=unix.cmxa gramlib.cmxa camlp_streams.cmxa
 TARGET=ledit.out
