Index: js_of_ocaml-3.8.0/compiler/lib/ocaml_version.ml
===================================================================
--- js_of_ocaml-3.8.0.orig/compiler/lib/ocaml_version.ml
+++ js_of_ocaml-3.8.0/compiler/lib/ocaml_version.ml
@@ -21,7 +21,7 @@ open! Stdlib
 type t = int list
 
 let split v =
-  match String.split_char ~sep:'+' v with
+  match String.split_char ~sep:'~' v with
   | [] -> assert false
   | x :: _ -> List.map (String.split_char ~sep:'.' x) ~f:int_of_string
 
Index: js_of_ocaml-3.8.0/compiler/ppx/ppx_optcomp_light.ml
===================================================================
--- js_of_ocaml-3.8.0.orig/compiler/ppx/ppx_optcomp_light.ml
+++ js_of_ocaml-3.8.0/compiler/ppx/ppx_optcomp_light.ml
@@ -53,7 +53,7 @@ end = struct
     split 0 0
 
   let split v =
-    match split_char ~sep:'+' v with
+    match split_char ~sep:'~' v with
     | [] -> assert false
     | x :: _ -> List.map (split_char ~sep:'.' x) ~f:int_of_string
 
