From 412ab74dc643afcb43a4c1604dd6d1708ec805a1 Mon Sep 17 00:00:00 2001
From: Kate <kit.ty.kate@disroot.org>
Date: Fri, 1 May 2020 18:54:00 +0100
Subject: [PATCH] OCaml 4.11 support

---
 compiler/lib/ocaml_compiler.ml | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/compiler/lib/ocaml_compiler.ml b/compiler/lib/ocaml_compiler.ml
index 17d5a73dd5..6175244952 100644
--- a/compiler/lib/ocaml_compiler.ml
+++ b/compiler/lib/ocaml_compiler.ml
@@ -24,7 +24,9 @@ let rec constant_of_const : _ -> Code.constant =
   function
   | Const_base (Const_int i) -> Int (Int32.of_int i)
   | Const_base (Const_char c) -> Int (Int32.of_int (Char.code c))
-  | Const_base (Const_string (s, _)) -> String s
+  | (Const_base (Const_string (s, _)) [@if ocaml_version < (4,11,0)])
+  | (Const_base (Const_string (s, _, _)) [@if ocaml_version >= (4,11,0)])
+    -> String s
   | Const_base (Const_float s) -> Float (float_of_string s)
   | Const_base (Const_int32 i) -> Int i
   | Const_base (Const_int64 i) -> Int64 i
