Index: merlin-4.6-414/src/config/gen_config.ml
===================================================================
--- merlin-4.6-414.orig/src/config/gen_config.ml
+++ merlin-4.6-414/src/config/gen_config.ml
@@ -17,5 +17,5 @@ let ocamlversion :
   | `OCaml_4_03_0 | `OCaml_4_04_0 | `OCaml_4_05_0 | `OCaml_4_06_0
   | `OCaml_4_07_0 | `OCaml_4_07_1 | `OCaml_4_08_0 | `OCaml_4_09_0
   | `OCaml_4_10_0 | `OCaml_4_11_0 | `OCaml_4_12_0 | `OCaml_4_13_0
-  | `OCaml_4_14_0 ] = %s
+  | `OCaml_4_14_0 | `OCaml_5_0_0] = %s
 |} ocaml_version_val
Index: merlin-4.6-414/src/utils/std.ml
===================================================================
--- merlin-4.6-414.orig/src/utils/std.ml
+++ merlin-4.6-414/src/utils/std.ml
@@ -450,11 +450,11 @@ module String = struct
      [_ascii] version. *)
   [@@@ocaml.warning "-3"]
 
-  let capitalize = capitalize
-  let uncapitalize = uncapitalize
+  let capitalize = capitalize_ascii
+  let uncapitalize = uncapitalize_ascii
 
-  let lowercase = lowercase
-  let uppercase = uppercase
+  let lowercase = lowercase_ascii
+  let uppercase = uppercase_ascii
 
   let split_on_char_ c s =
     match String.index s c with
@@ -620,8 +620,8 @@ module Char = struct
   [@@@ocaml.warning "-3"]
 
   include Char
-  let is_lowercase c = lowercase c = c
-  let is_uppercase c = uppercase c = c
+  let is_lowercase c = lowercase_ascii c = c
+  let is_uppercase c = uppercase_ascii c = c
   let is_strictly_lowercase c = not (is_uppercase c)
   let is_strictly_uppercase c = not (is_lowercase c)
 end
