Index: gettext-v0.4.1/test/common/dune
===================================================================
--- gettext-v0.4.1.orig/test/common/dune
+++ gettext-v0.4.1/test/common/dune
@@ -1,3 +1,3 @@
 (library
  (name common)
- (libraries gettext.base oUnit))
+ (libraries gettext.base ounit2))
Index: gettext-v0.4.1/test/dune
===================================================================
--- gettext-v0.4.1.orig/test/dune
+++ gettext-v0.4.1/test/dune
@@ -6,7 +6,7 @@
   ../src/bin/ocaml-xgettext/xgettext.exe
   (glob_files testdata/*)
   (glob_files testdata/fr_FR/LC_MESSAGES/*))
- (libraries oUnit str fileutils gettext.extension common)
+ (libraries ounit2 str fileutils gettext.extension common)
  (action
   (run %{test}
     -runner sequential
Index: gettext-v0.4.1/test/test-camomile/dune
===================================================================
--- gettext-v0.4.1.orig/test/test-camomile/dune
+++ gettext-v0.4.1/test/test-camomile/dune
@@ -4,6 +4,6 @@
  (deps
   (glob_files testdata/*)
   (glob_files ../testdata/fr_FR/LC_MESSAGES/*))
- (libraries common gettext-camomile oUnit fileutils)
+ (libraries common gettext-camomile ounit2 fileutils)
  (action
   (run %{test} -test-dir ../testdata)))
Index: gettext-v0.4.1/test/test-stub/dune
===================================================================
--- gettext-v0.4.1.orig/test/test-stub/dune
+++ gettext-v0.4.1/test/test-stub/dune
@@ -4,6 +4,6 @@
  (deps
   (glob_files ../testdata/*)
   (glob_files ../testdata/fr_FR/LC_MESSAGES/*))
- (libraries common gettext-stub oUnit)
+ (libraries common gettext-stub ounit2)
  (action
   (run %{test} -test-dir ../testdata)))
