Index: num-1.1/src/Makefile
===================================================================
--- num-1.1.orig/src/Makefile
+++ num-1.1/src/Makefile
@@ -84,9 +84,9 @@ TOINSTALL_STUBS=dllnums.$(SO)
 
 install:
 	$(OCAMLFIND) install num META
-	$(INSTALL_DATA) $(TOINSTALL) $(STDLIBDIR)
+	$(INSTALL_DATA) $(TOINSTALL) $(DESTDIR)$(STDLIBDIR)
 ifeq "$(SUPPORTS_SHARED_LIBRARIES)" "true"
-	$(INSTALL_DLL) $(TOINSTALL_STUBS) $(STDLIBDIR)/stublibs
+	$(INSTALL_DLL) $(TOINSTALL_STUBS) $(DESTDIR)$(STDLIBDIR)/stublibs
 endif
 
 uninstall:
