Index: ppx_expect-0.11.0/evaluator/ppx_expect_evaluator.ml
===================================================================
--- ppx_expect-0.11.0.orig/evaluator/ppx_expect_evaluator.ml
+++ ppx_expect-0.11.0/evaluator/ppx_expect_evaluator.ml
@@ -148,7 +148,7 @@ let process_group ~use_color ~in_place ~
             ~temp_dir:(Caml.Filename.dirname filename)
         in
         write_corrected ~file:tmp_corrected;
-        Print_diff.print ~file1:filename ~file2:tmp_corrected ~use_color ?diff_command ();
+        Ppxlib_print_diff.print ~file1:filename ~file2:tmp_corrected ~use_color ?diff_command ();
         Caml.Sys.rename tmp_corrected dot_corrected;
         Failure
 ;;
