Index: fmt-0.8.8/_tags
===================================================================
--- fmt-0.8.8.orig/_tags
+++ fmt-0.8.8/_tags
@@ -1,4 +1,4 @@
-true : bin_annot, safe_string, package(seq), package(stdlib-shims)
+true : bin_annot, safe_string, package(stdlib-shims)
 <_b0> : -traverse
 <src> : include
 <src/fmt_tty*> : package(unix)
Index: fmt-0.8.8/pkg/META
===================================================================
--- fmt-0.8.8.orig/pkg/META
+++ fmt-0.8.8/pkg/META
@@ -1,6 +1,6 @@
 description = "OCaml Format pretty-printer combinators"
 version = "0.8.8"
-requires = "seq stdlib-shims"
+requires = "stdlib-shims"
 archive(byte) = "fmt.cma"
 archive(native) = "fmt.cmxa"
 plugin(byte) = "fmt.cma"
