Index: fmt-0.8.6/_tags
===================================================================
--- fmt-0.8.6.orig/_tags
+++ fmt-0.8.6/_tags
@@ -1,4 +1,4 @@
-true : bin_annot, safe_string, package(seq), package(stdlib-shims)
+true : bin_annot, safe_string
 
 <src> : include
 <src/fmt_tty*> : package(unix)
Index: fmt-0.8.6/pkg/META
===================================================================
--- fmt-0.8.6.orig/pkg/META
+++ fmt-0.8.6/pkg/META
@@ -1,6 +1,6 @@
 description = "OCaml Format pretty-printer combinators"
 version = "0.8.6"
-requires = "seq stdlib-shims"
+requires = ""
 archive(byte) = "fmt.cma"
 archive(native) = "fmt.cmxa"
 plugin(byte) = "fmt.cma"
