Index: pxp-1.2.9/tools/src/odoc/Makefile
===================================================================
--- pxp-1.2.9.orig/tools/src/odoc/Makefile
+++ pxp-1.2.9/tools/src/odoc/Makefile
@@ -3,7 +3,7 @@ include $(TOP_DIR)/Makefile.rules
 
 SRC = chtml.ml
 OBJ = $(SRC:.ml=.cmo)
-OCAMLC_OPTIONS += -I +ocamldoc
+OCAMLC_OPTIONS += -I +ocamldoc -I +compiler-libs
 CLEAN_LIST += chtml.ml
 
 .PHONY: all
Index: pxp-1.2.9/tools/src/odoc/chtml.ml
===================================================================
--- pxp-1.2.9.orig/tools/src/odoc/chtml_ocaml4.ml
+++ pxp-1.2.9/tools/src/odoc/chtml_ocaml4.ml
@@ -34,7 +34,7 @@ open Printf
 open Odoc_info
 open Module
 
-module StringSet = Odoc_html.StringSet
+module StringSet = Odoc_html.String.Set
 
 
 let word_re = Str.regexp "[ \t\r\n]+"
Index: pxp-1.2.9/src/pxp-engine/pxp_lexing.mli
===================================================================
--- pxp-1.2.9.orig/src/pxp-engine/pxp_lexing.mli
+++ pxp-1.2.9/src/pxp-engine/pxp_lexing.mli
@@ -10,11 +10,11 @@
 
 type lexbuf = Lexing.lexbuf
 
-val from_channel : in_channel -> lexbuf
+val from_channel : ?with_positions:bool -> in_channel -> lexbuf
 
-val from_string : string -> lexbuf
+val from_string : ?with_positions:bool -> string -> lexbuf
 
-val from_function : (Bytes.t -> int -> int) -> lexbuf
+val from_function : ?with_positions:bool -> (Bytes.t -> int -> int) -> lexbuf
 
 val lexeme : lexbuf -> string
 
