Index: coccinelle-1.1.1/Makefile
===================================================================
--- coccinelle-1.1.1.orig/Makefile
+++ coccinelle-1.1.1/Makefile
@@ -98,7 +98,7 @@ PREFIX_spatch :=
 
 PREFIX_spgen := tools/spgen/source/
 
-CORE_LIBS := unix bigarray str \
+CORE_LIBS := unix str \
 	$(STDCOMPATDIR)/stdcompat \
 	$(patsubst %,bytes,$(BYTESDIR)) \
 	$(patsubst %,pcre,$(filter %/pcre.cma,$(LNKLIBS)))
Index: coccinelle-1.1.1/configure.ac
===================================================================
--- coccinelle-1.1.1.orig/configure.ac
+++ coccinelle-1.1.1/configure.ac
@@ -147,7 +147,6 @@ AC_COCCI_INIT_PKG_EMPTY([parmap])
 dnl  required ocaml packages
 AC_MSG_NOTICE([the following OCaml packages should be provided by your ocaml installation])
 AC_REQ_COCCI_STDPKG([unix])
-AC_REQ_COCCI_STDPKG([bigarray])
 AC_REQ_COCCI_STDPKG([str])
 
 dnl  required modules (Some are available locally too)
