Index: coccinelle-1.0.7/Makefile
===================================================================
--- coccinelle-1.0.7.orig/Makefile
+++ coccinelle-1.0.7/Makefile
@@ -186,7 +186,11 @@ SEARCH_PATHS := \
 	commons/ocamlextra $(LIBRARIES) $(PREFIX_spgen) $(PCREDIR) $(PYMLDIR) \
 	$(PARMAPDIR) $(BYTESDIR) $(STDCOMPATDIR)
 
+ODEP_SEARCH_PATHS := \
+	commons/ocamlextra $(LIBRARIES) $(PREFIX_spgen)
+
 SEARCH_PATH_FLAGS := $(addprefix -I ,$(SEARCH_PATHS))
+ODEP_SEARCH_PATH_FLAGS := $(addprefix -I ,$(ODEP_SEARCH_PATHS))
 
 CCLIB := $(patsubst %,-cclib -lparmap_stubs,$(filter %/parmap.cma,$(LNKLIBS))) \
 	-cclib -L.
@@ -195,7 +199,7 @@ OCAMLC_CMD := $(OCAMLC) $(SEARCH_PATH_FL
 
 OCAMLOPT_CMD := $(OCAMLOPT) $(SEARCH_PATH_FLAGS) $(CCLIB) $(EXTRA_OCAML_FLAGS)
 
-OCAMLDEP_CMD := $(OCAMLDEP) $(SEARCH_PATH_FLAGS) \
+OCAMLDEP_CMD := $(OCAMLDEP) $(ODEP_SEARCH_PATH_FLAGS) \
 	$(addprefix -ml-synonym ,.mll .mly) \
 	$(addprefix -mli-synonym ,.ml .mll .mly)
 
