Index: optcomp-1.6/src/optcomp.ml
===================================================================
--- optcomp-1.6.orig/src/optcomp.ml
+++ optcomp-1.6/src/optcomp.ml
@@ -86,7 +86,7 @@ let rec print mode current_fname current
                 (* Read the part to copy. *)
                 let str = String.create len in
                 really_input ic str 0 len;
-                (str, Loc.stop_line loc, Loc.stop_off loc - Loc.stop_bol loc)
+                (Bytes.to_string str, Loc.stop_line loc, Loc.stop_off loc - Loc.stop_bol loc)
         in
         if current_fname = fname && current_line = line && current_col = col then
           (* If we at the right position, just print the string. *)
