Index: ocaml-graphql-server-0.14.0/graphql-async/src/graphql_async.ml
===================================================================
--- ocaml-graphql-server-0.14.0.orig/graphql-async/src/graphql_async.ml
+++ ocaml-graphql-server-0.14.0/graphql-async/src/graphql_async.ml
@@ -10,7 +10,7 @@ module Schema =
 
         let map t f =
           Async_kernel.Pipe.map' t ~f:(fun q ->
-              Async_kernel.Deferred.Queue.map q ~f)
+              Async_kernel.Deferred.Queue.map q ~f ~how:`Sequential)
 
         let iter t f = Async_kernel.Pipe.iter t ~f
 
