Index: ocaml-4.08.0-beta1/configure.ac
===================================================================
--- ocaml-4.08.0-beta1.orig/configure.ac
+++ ocaml-4.08.0-beta1/configure.ac
@@ -598,6 +598,9 @@ AS_CASE([$CC,$host],
     AC_DEFINE([HAS_ARCH_CODE32], [1])
 )
 
+# Append user flags
+common_cflags="${common_cflags} ${CFLAGS}"
+oc_ldflags="${oc_ldflags} ${LDFLAGS}"
 
 ## Program to use to install files
 AC_PROG_INSTALL
@@ -700,7 +703,7 @@ AS_IF([test x"$enable_shared" != "xno"],
     [[*-*-linux*|*-*-freebsd[3-9]*|*-*-freebsd[1-9][0-9]*\
     |*-*-openbsd*|*-*-netbsd*|*-*-dragonfly*|*-*-gnu*|*-*-haiku*]],
       [sharedlib_cflags="-fPIC"
-      mksharedlib="$CC -shared"
+      mksharedlib="$CC ${LDFLAGS} -shared"
       oc_ldflags="$oc_ldflags -Wl,-E"
       rpath="-Wl,-rpath,"
       mksharedlibrpath="-Wl,-rpath,"
