Index: ocsigenserver-5.0.0/src/server/ocsigen_cohttp.ml
===================================================================
--- ocsigenserver-5.0.0.orig/src/server/ocsigen_cohttp.ml
+++ ocsigenserver-5.0.0/src/server/ocsigen_cohttp.ml
@@ -32,7 +32,7 @@ let _print_request fmt request =
        (print_list
           (fun fmt value -> Format.fprintf fmt "\t%s = %s\n" key value)
           fmt
-          values))
+          [values]))
     (Cohttp.Request.headers request)
 
 let connections = Hashtbl.create 256
