Index: ocsigenserver-2.9/configure
===================================================================
--- ocsigenserver-2.9.orig/configure
+++ ocsigenserver-2.9/configure
@@ -412,7 +412,6 @@ check_library lwt "See: http://ocsigen.o
 check_library lwt.unix  "Missing support for 'unix' in lwt."
 check_library lwt_react   "See: http://ocsigen.org/lwt"
 check_library lwt_ssl   "See: http://ocsigen.org/lwt"
-check_library lwt.preemptive "Missing support for 'preemptive' in lwt."
 
 check_library netstring \
     "See ocamlnet: http://projects.camlcity.org/projects/ocamlnet.html"
@@ -462,17 +461,6 @@ if [ "$with_camlzip" -gt 0 ]; then
     fi
 fi
 
-# Check Lwt.preemptive
-if [ "$with_preempt" -gt 0 ]; then
-    if test_library lwt.preemptive; then
-	echo -n
-    elif [ "$with_preempt" -gt 1 ]; then
-	fail_library lwt.preemptive "Missing support for 'preemptive' in lwt."
-    else
-	with_preempt=0
-    fi
-fi
-
 if [ "$with_sqlite" -eq 1 ] && [ "$with_preempt" -eq -1 ]; then
     echo "preemptive threads are needed by sqlite, enable it with -with-preempt"
     exit 1
Index: ocsigenserver-2.9/Makefile.options
===================================================================
--- ocsigenserver-2.9.orig/Makefile.options
+++ ocsigenserver-2.9/Makefile.options
@@ -25,7 +25,7 @@ endif
 ## but also to generate src/baselib/ocsigen_config.ml and src/files/META
 
 ifeq "$(PREEMPTIVE)" "YES"
-LWT_PREEMPTIVE_PACKAGE:=lwt.preemptive
+LWT_PREEMPTIVE_PACKAGE:=lwt.unix
 endif
 
 BASE_PACKAGE := lwt ipaddr bytes
@@ -33,6 +33,7 @@ BASE_PACKAGE := lwt ipaddr bytes
 SERVER_PACKAGE := lwt_ssl           \
 		  bytes			\
 	          ${LWT_PREEMPTIVE_PACKAGE} \
+	          lwt_log 	    \
                   ipaddr            \
 	          netstring         \
 	          netstring-pcre    \
Index: ocsigenserver-2.9/src/baselib/Makefile
===================================================================
--- ocsigenserver-2.9.orig/src/baselib/Makefile
+++ ocsigenserver-2.9/src/baselib/Makefile
@@ -9,6 +9,7 @@ PACKAGE  := \
 	findlib \
 	tyxml \
 	${LWT_PREEMPTIVE_PACKAGE} \
+	lwt_log \
 	ipaddr
 
 LIBS     := ${addprefix -package ,${PACKAGE}}
Index: ocsigenserver-2.9/src/baselib/ocsigen_stream.ml
===================================================================
--- ocsigenserver-2.9.orig/src/baselib/ocsigen_stream.ml
+++ ocsigenserver-2.9/src/baselib/ocsigen_stream.ml
@@ -229,10 +229,10 @@ let of_file filename =
   let fd = Lwt_unix.of_unix_file_descr
       (Unix.openfile filename [Unix.O_RDONLY;Unix.O_NONBLOCK] 0o666)
   in
-  let ch = Lwt_chan.in_channel_of_descr fd in
+  let ch = Lwt_io.of_fd Lwt_io.Input fd in
   let buf = Bytes.create 1024 in
   let rec aux () =
-    Lwt_chan.input ch buf 0 1024 >>= fun n ->
+    Lwt_io.read_into ch buf 0 1024 >>= fun n ->
     if n = 0 then empty None else
       (* Streams should be immutable, thus we always make a copy
          of the buffer *)
Index: ocsigenserver-2.9/src/http/Makefile
===================================================================
--- ocsigenserver-2.9.orig/src/http/Makefile
+++ ocsigenserver-2.9/src/http/Makefile
@@ -4,6 +4,7 @@ PACKAGE  := \
 	netstring \
 	netstring-pcre \
 	lwt_ssl \
+	lwt_log \
 	tyxml
 
 LIBS     := -I ../baselib ${addprefix -package ,${PACKAGE}}
Index: ocsigenserver-2.9/src/http/ocsigen_http_com.mli
===================================================================
--- ocsigenserver-2.9.orig/src/http/ocsigen_http_com.mli
+++ ocsigenserver-2.9/src/http/ocsigen_http_com.mli
@@ -69,7 +69,7 @@ val wait_all_senders : connection -> uni
    interrupted stream exception.
 *)
 val write_stream :
-  ?chunked:bool -> Lwt_chan.out_channel -> string Ocsigen_stream.t -> unit Lwt.t
+  ?chunked:bool -> Lwt_io.output_channel -> string Ocsigen_stream.t -> unit Lwt.t
 
 (****)
 
Index: ocsigenserver-2.9/src/extensions/Makefile
===================================================================
--- ocsigenserver-2.9.orig/src/extensions/Makefile
+++ ocsigenserver-2.9/src/extensions/Makefile
@@ -6,6 +6,7 @@ PACKAGE  :=		\
 	ipaddr		\
 	lwt_ssl		\
 	lwt_react	\
+	lwt_log		\
 	netstring	\
 	netstring-pcre	\
 	tyxml.parser
Index: ocsigenserver-2.9/src/extensions/cgimod.ml
===================================================================
--- ocsigenserver-2.9.orig/src/extensions/cgimod.ml
+++ ocsigenserver-2.9/src/extensions/cgimod.ml
@@ -368,7 +368,7 @@ let recupere_cgi head re doc_root filena
     Lwt_timeout.start timeout;
 
     (* A thread giving POST data to the CGI script: *)
-    let post_in_ch = Lwt_chan.out_channel_of_descr post_in in
+    let post_in_ch = Lwt_io.of_fd Lwt_io.Output post_in in
     ignore
       (catch
          (fun () ->
@@ -376,7 +376,7 @@ let recupere_cgi head re doc_root filena
              | None -> Lwt_unix.close post_in
              | Some content_post ->
                Ocsigen_http_com.write_stream post_in_ch content_post >>= fun () ->
-               Lwt_chan.flush post_in_ch >>= fun () ->
+               Lwt_io.flush post_in_ch >>= fun () ->
                Lwt_unix.close post_in
             ))
          (*XXX Check possible errors! *)
@@ -391,9 +391,9 @@ let recupere_cgi head re doc_root filena
 
     (* A thread listening the error output of the CGI script
        and writing them in warnings.log *)
-    let err_channel = Lwt_chan.in_channel_of_descr err_out in
+    let err_channel = Lwt_io.of_fd Lwt_io.Input err_out in
     let rec get_errors () =
-      Lwt_chan.input_line err_channel >>= fun err ->
+      Lwt_io.read_line err_channel >>= fun err ->
       Lwt_log.ign_warning ~section err;
       get_errors ()
     in ignore
Index: ocsigenserver-2.9/src/extensions/ocsipersist-dbm/Makefile
===================================================================
--- ocsigenserver-2.9.orig/src/extensions/ocsipersist-dbm/Makefile
+++ ocsigenserver-2.9/src/extensions/ocsipersist-dbm/Makefile
@@ -1,7 +1,7 @@
 include ../../../Makefile.config
 
 PACKAGE  := ${LWT_PREEMPTIVE_PACKAGE} \
-	    lwt.unix     \
+	    lwt.unix lwt_log     \
 	    tyxml.parser \
 	    dbm          \
 
Index: ocsigenserver-2.9/src/extensions/ocsipersist-dbm/ocsidbm.ml
===================================================================
--- ocsigenserver-2.9.orig/src/extensions/ocsipersist-dbm/ocsidbm.ml
+++ ocsigenserver-2.9/src/extensions/ocsipersist-dbm/ocsidbm.ml
@@ -198,8 +198,8 @@ let _ = Unix.setsid ()
 (** Communication functions: *)
 
 let send outch v =
-  Lwt_chan.output_value outch v >>=
-  (fun () -> Lwt_chan.flush outch)
+  Lwt_io.write_value outch v >>=
+  (fun () -> Lwt_io.flush outch)
 
 let execute outch =
   let handle_errors f = try f () with e -> send outch (Error e) in
@@ -241,7 +241,7 @@ let execute outch =
 let nb_clients = ref 0
 
 let rec listen_client inch outch =
-  Lwt_chan.input_value inch >>=
+  Lwt_io.read_value inch >>=
   (fun v -> execute outch v) >>=
   (fun () -> listen_client inch outch)
 
@@ -260,8 +260,8 @@ let rec loop socket =
      ignore (
        b := true;
        nb_clients := !nb_clients + 1;
-       let inch = Lwt_chan.in_channel_of_descr indescr in
-       let outch = Lwt_chan.out_channel_of_descr indescr in
+       let inch = Lwt_io.of_fd Lwt_io.Input indescr in
+       let outch = Lwt_io.of_fd Lwt_io.Output indescr in
        catch
          (fun () -> listen_client inch outch >>= finish)
          finish);
Index: ocsigenserver-2.9/src/extensions/ocsipersist-dbm/ocsipersist.ml
===================================================================
--- ocsigenserver-2.9.orig/src/extensions/ocsipersist-dbm/ocsipersist.ml
+++ ocsigenserver-2.9/src/extensions/ocsipersist-dbm/ocsipersist.ml
@@ -151,12 +151,12 @@ let init_fun config =
        Lwt_log.ign_warning ~section "Initializing ...");
   let indescr = get_indescr 2 in
   if delay_loading then (
-    inch  := (indescr >>= fun r -> return (Lwt_chan.in_channel_of_descr r));
-    outch := (indescr >>= fun r -> return (Lwt_chan.out_channel_of_descr r));
+    inch  := (indescr >>= fun r -> return (Lwt_io.of_fd Lwt_io.Input r));
+    outch := (indescr >>= fun r -> return (Lwt_io.of_fd Lwt_io.Output r));
   ) else (
     let r = Lwt_main.run indescr in
-    inch  := return (Lwt_chan.in_channel_of_descr r);
-    outch := return (Lwt_chan.out_channel_of_descr r);
+    inch  := return (Lwt_io.of_fd Lwt_io.Input r);
+    outch := return (Lwt_io.of_fd Lwt_io.Output r);
     Lwt_log.ign_warning ~section "...Initialization complete";
   )
 
@@ -171,9 +171,9 @@ let send =
        !inch >>= fun inch ->
        !outch >>= fun outch ->
        previous :=
-         (Lwt_chan.output_value outch v >>= fun () ->
-          Lwt_chan.flush outch >>= fun () ->
-          Lwt_chan.input_value inch);
+         (Lwt_io.write_value outch v >>= fun () ->
+          Lwt_io.flush outch >>= fun () ->
+          Lwt_io.read_value inch);
        !previous)
 
 let db_get (store, name) =
Index: ocsigenserver-2.9/src/extensions/ocsipersist-pgsql/Makefile
===================================================================
--- ocsigenserver-2.9.orig/src/extensions/ocsipersist-pgsql/Makefile
+++ ocsigenserver-2.9/src/extensions/ocsipersist-pgsql/Makefile
@@ -1,6 +1,6 @@
 include ../../../Makefile.config
 
-PACKAGE  := tyxml.parser pgocaml lwt
+PACKAGE  := tyxml.parser pgocaml lwt lwt_log
 
 LIBS     := -I ../../baselib -I ../../http -I ../../server \
 	    ${addprefix -package ,${PACKAGE}}
Index: ocsigenserver-2.9/src/extensions/ocsipersist-sqlite/Makefile
===================================================================
--- ocsigenserver-2.9.orig/src/extensions/ocsipersist-sqlite/Makefile
+++ ocsigenserver-2.9/src/extensions/ocsipersist-sqlite/Makefile
@@ -1,6 +1,6 @@
 include ../../../Makefile.config
 
-PACKAGE  := lwt.preemptive  \
+PACKAGE  := lwt lwt_log \
 	    tyxml.parser \
 	    sqlite3      \
 
Index: ocsigenserver-2.9/src/server/ocsigen_server.ml
===================================================================
--- ocsigenserver-2.9.orig/src/server/ocsigen_server.ml
+++ ocsigenserver-2.9/src/server/ocsigen_server.ml
@@ -1317,13 +1317,13 @@ let start_server () =
 
       Ocsigen_extensions.end_initialisation ();
 
-      let pipe = Lwt_chan.in_channel_of_descr
+      let pipe = Lwt_io.of_fd Lwt_io.Input
           (Lwt_unix.of_unix_file_descr
              (Unix.openfile commandpipe
                 [Unix.O_RDWR; Unix.O_NONBLOCK; Unix.O_APPEND] 0o660)) in
 
       let rec f () =
-        Lwt_chan.input_line pipe >>= fun s ->
+        Lwt_io.read_line pipe >>= fun s ->
         Lwt_log.ign_notice ~section ("Command received: "^s);
         (Lwt.catch
            (fun () ->
