Index: ocsigenserver-2.8/src/extensions/Makefile
===================================================================
--- ocsigenserver-2.8.orig/src/extensions/Makefile
+++ ocsigenserver-2.8/src/extensions/Makefile
@@ -11,8 +11,8 @@ PACKAGE  :=		\
 	tyxml.parser
 
 LIBS     := -I ../baselib -I ../http -I ../server ${addprefix -package ,${PACKAGE}}
-OCAMLC   := $(OCAMLFIND) ocamlc ${BYTEDBG}
-OCAMLOPT := $(OCAMLFIND) ocamlopt ${OPTDBG}
+OCAMLC   := $(OCAMLFIND) ocamlc ${BYTEDBG} ${THREAD}
+OCAMLOPT := $(OCAMLFIND) ocamlopt ${OPTDBG} ${THREAD}
 OCAMLDOC := $(OCAMLFIND) ocamldoc
 OCAMLDEP := $(OCAMLFIND) ocamldep
 
Index: ocsigenserver-2.8/src/http/Makefile
===================================================================
--- ocsigenserver-2.8.orig/src/http/Makefile
+++ ocsigenserver-2.8/src/http/Makefile
@@ -7,8 +7,8 @@ PACKAGE  := \
 	tyxml
 
 LIBS     := -I ../baselib ${addprefix -package ,${PACKAGE}}
-OCAMLC   := $(OCAMLFIND) ocamlc ${BYTEDBG}
-OCAMLOPT := $(OCAMLFIND) ocamlopt ${OPTDBG}
+OCAMLC   := $(OCAMLFIND) ocamlc ${BYTEDBG} ${THREAD}
+OCAMLOPT := $(OCAMLFIND) ocamlopt ${OPTDBG} ${THREAD}
 OCAMLDOC := $(OCAMLFIND) ocamldoc
 OCAMLDEP := $(OCAMLFIND) ocamldep
 
