Index: ocurl-0.9.0/curl-helper.c
===================================================================
--- ocurl-0.9.0.orig/curl-helper.c
+++ ocurl-0.9.0/curl-helper.c
@@ -5,6 +5,10 @@
  ***  Copyright (c) 2009, ygrek, <ygrek@autistici.org>
  ***/
 
+#ifndef CAML_NAME_SPACE
+#define CAML_NAME_SPACE
+#endif
+
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
@@ -17,9 +21,6 @@
 #define CURL_DISABLE_TYPECHECK
 #include <curl/curl.h>
 
-#ifndef CAML_NAME_SPACE
-#define CAML_NAME_SPACE
-#endif
 #include <caml/alloc.h>
 #include <caml/memory.h>
 #include <caml/mlvalues.h>
