Index: ocsigen-toolkit-2.4.1/src/widgets/ot_picture_uploader.eliom
===================================================================
--- ocsigen-toolkit-2.4.1.orig/src/widgets/ot_picture_uploader.eliom
+++ ocsigen-toolkit-2.4.1/src/widgets/ot_picture_uploader.eliom
@@ -424,7 +424,7 @@ let%server mk_service name arg_deriver =
           Eliom_parameter.(
             ocaml "service_arg" arg_deriver **
             ocaml "cropping"
-              [%derive.json: (float * float * float * float) option ]
+              [%json: (float * float * float * float) option ]
             ** file "f")))
     ()
 
