Index: ocsigen-toolkit-1.1.0/Makefile.options
===================================================================
--- ocsigen-toolkit-1.1.0.orig/Makefile.options
+++ ocsigen-toolkit-1.1.0/Makefile.options
@@ -40,9 +40,9 @@ TEMPLATE_DIR := template.distillery
 TEMPLATE_NAME := none.pgocaml
 
 # OCamlfind packages for the server
-SERVER_PACKAGES := calendar lwt.ppx js_of_ocaml-ppx.deriving
+SERVER_PACKAGES := calendar lwt_ppx js_of_ocaml-ppx.deriving
 # OCamlfind packages for the client
-CLIENT_PACKAGES := calendar js_of_ocaml js_of_ocaml-ppx lwt.ppx js_of_ocaml-ppx.deriving
+CLIENT_PACKAGES := calendar js_of_ocaml js_of_ocaml-ppx lwt_ppx js_of_ocaml-ppx.deriving
 
 # Debug package (yes/no): Debugging info in compilation
 DEBUG := yes
