Index: ocamldap-2.4.2/src/ldap/dune
===================================================================
--- ocamldap-2.4.2.orig/src/ldap/dune
+++ ocamldap-2.4.2/src/ldap/dune
@@ -6,4 +6,4 @@
   (public_name ldap)
   (wrapped false)
   (modules_without_implementation ldap_urlparser)
-  (libraries str pcre ssl))
+  (libraries str pcre ssl camlp-streams))
Index: ocamldap-2.4.2/dune-project
===================================================================
--- ocamldap-2.4.2.orig/dune-project
+++ ocamldap-2.4.2/dune-project
@@ -19,6 +19,7 @@
     (ocaml (>= 4.03.0))
     (ocamlnet (>= 3.6.0))
     pcre
+    camlp-streams
     (ssl (>= 0.5.3)))
   (conflicts
     (ocamldap (<> transition)))
