Index: ocaml-4.08.0-beta1/tools/Makefile
===================================================================
--- ocaml-4.08.0-beta1.orig/tools/Makefile
+++ ocaml-4.08.0-beta1/tools/Makefile
@@ -322,7 +322,7 @@ DEF_SYMBOL_PREFIX = '-Dsymbol_prefix=""'
 endif
 
 objinfo_helper$(EXE): objinfo_helper.c $(ROOTDIR)/runtime/caml/s.h
-	$(CC) $(OC_CFLAGS) $(OC_CPPFLAGS) -I$(ROOTDIR)/runtime $(OUTPUTEXE)$@ \
+	$(CC) $(OC_CFLAGS) $(OC_CPPFLAGS) $(OC_LDFLAGS) -I$(ROOTDIR)/runtime $(OUTPUTEXE)$@ \
           $(DEF_SYMBOL_PREFIX) $(LIBBFD_INCLUDE) $< $(LIBBFD_LINK)
 
 OBJINFO=$(ROOTDIR)/compilerlibs/ocamlcommon.cma \
