Index: qcheck-0.10/example/ounit/dune
===================================================================
--- qcheck-0.10.orig/example/ounit/dune
+++ qcheck-0.10/example/ounit/dune
@@ -1,5 +1,5 @@
 
 (executables
   (names QCheck_ounit_test QCheck_test)
-  (libraries qcheck oUnit qcheck-ounit)
+  (libraries qcheck ounit2 qcheck-ounit)
   )
Index: qcheck-0.10/src/ounit/dune
===================================================================
--- qcheck-0.10.orig/src/ounit/dune
+++ qcheck-0.10/src/ounit/dune
@@ -4,6 +4,6 @@
   (public_name qcheck-ounit)
   (optional)
   (wrapped false)
-  (libraries unix bytes qcheck-core qcheck-core.runner oUnit)
+  (libraries unix bytes qcheck-core qcheck-core.runner ounit2)
   (flags :standard -w +a-4-42-44-48-50-58-32-60@8 -safe-string)
   )
