Index: ppx_sexp_conv-0.15.0/expander/expand_sexp_of.ml
===================================================================
--- ppx_sexp_conv-0.15.0.orig/expander/expand_sexp_of.ml
+++ ppx_sexp_conv-0.15.0/expander/expand_sexp_of.ml
@@ -727,7 +727,7 @@ module Str_generate_sexp_of = struct
     let loc = ec.ptyexn_loc in
     let expr =
       match ec.ptyexn_constructor with
-      | { pext_name = cnstr; pext_kind = Pext_decl (extension_constructor_kind, None); _ }
+      | { pext_name = cnstr; pext_kind = Pext_decl (_, extension_constructor_kind, None); _ }
         ->
         let constr_lid = Located.map lident cnstr in
         branch_sum
@@ -748,7 +748,7 @@ module Str_generate_sexp_of = struct
               Conversion.to_expression
                 ~loc
                 (Conversion.of_lambda [ converter; assert_false ])]]
-      | { pext_kind = Pext_decl (_, Some _); _ } ->
+      | { pext_kind = Pext_decl (_, _, Some _); _ } ->
         Location.raise_errorf ~loc "sexp_of_exn/:"
       | { pext_kind = Pext_rebind _; _ } ->
         Location.raise_errorf ~loc "sexp_of_exn/rebind"
