Index: ocaml-snappy-0.1.0/test.ml
===================================================================
--- ocaml-snappy-0.1.0.orig/test.ml
+++ ocaml-snappy-0.1.0/test.ml
@@ -4,12 +4,12 @@ let () =
   let check n =
     let s = String.create n in
     for i = 0 to pred n do s.[i] <- Char.chr (Random.int 256) done;
-    let c = Snappy.compress s in
+    let c = Snappy.compress (Bytes.to_string s) in
     assert (true = Snappy.is_valid c);
-    assert (false = Snappy.is_valid s);
-    assert (String.length s = Snappy.get_uncompressed_size c);
+    assert (false = Snappy.is_valid (Bytes.to_string s));
+    assert (Bytes.length s = Snappy.get_uncompressed_size c);
 (*     assert (try let (_:string) = Snappy.uncompress s in false with Snappy.Error _ -> true); *)
-    assert (s = Snappy.uncompress c)
+    assert (Bytes.to_string s = Snappy.uncompress c)
   in
   for n = 0 to 1024 do check n done;
   for n = 0 to 1024 do check (Random.int 16384 + 16384) done;
