Index: merlin-3.3.2/src/config/gen_config.ml
===================================================================
--- merlin-3.3.2.orig/src/config/gen_config.ml
+++ merlin-3.3.2/src/config/gen_config.ml
@@ -15,5 +15,5 @@ let version = "%%VERSION%%"
 let ocamlversion :
   [ `OCaml_4_02_0 | `OCaml_4_02_1 | `OCaml_4_02_2 | `OCaml_4_02_3
   | `OCaml_4_03_0 | `OCaml_4_04_0 | `OCaml_4_05_0 | `OCaml_4_06_0
-  | `OCaml_4_07_0 | `OCaml_4_07_1 | `OCaml_4_08_0 ] = %s
+  | `OCaml_4_07_0 | `OCaml_4_07_1 | `OCaml_4_08_0 | `OCaml_4_09_0 ] = %s
 |} ocaml_version_val
Index: merlin-3.3.2/src/ocaml/typing/409/typecore.ml
===================================================================
--- merlin-3.3.2.orig/src/ocaml/typing/409/typecore.ml
+++ merlin-3.3.2/src/ocaml/typing/409/typecore.ml
@@ -3764,28 +3764,30 @@ and type_format loc str env =
         | Int_u  -> mk_constr "Int_u"  [] | Int_Cd -> mk_constr "Int_Cd" []
         | Int_Ci -> mk_constr "Int_Ci" [] | Int_Cu -> mk_constr "Int_Cu" []
       and mk_fconv fconv = match fconv with
-        | Float_f  -> mk_constr "Float_f"  []
-        | Float_pf -> mk_constr "Float_pf" []
-        | Float_sf -> mk_constr "Float_sf" []
-        | Float_e  -> mk_constr "Float_e"  []
-        | Float_pe -> mk_constr "Float_pe" []
-        | Float_se -> mk_constr "Float_se" []
-        | Float_E  -> mk_constr "Float_E"  []
-        | Float_pE -> mk_constr "Float_pE" []
-        | Float_sE -> mk_constr "Float_sE" []
-        | Float_g  -> mk_constr "Float_g"  []
-        | Float_pg -> mk_constr "Float_pg" []
-        | Float_sg -> mk_constr "Float_sg" []
-        | Float_G  -> mk_constr "Float_G"  []
-        | Float_pG -> mk_constr "Float_pG" []
-        | Float_sG -> mk_constr "Float_sG" []
-        | Float_h  -> mk_constr "Float_h"  []
-        | Float_ph -> mk_constr "Float_ph" []
-        | Float_sh -> mk_constr "Float_sh" []
-        | Float_H  -> mk_constr "Float_H"  []
-        | Float_pH -> mk_constr "Float_pH" []
-        | Float_sH -> mk_constr "Float_sH" []
-        | Float_F  -> mk_constr "Float_F"  []
+        | Float_flag_, Float_f  -> mk_constr "Float_f"  []
+        | Float_flag_p, Float_f -> mk_constr "Float_pf" []
+        | Float_flag_s, Float_f -> mk_constr "Float_sf" []
+        | Float_flag_, Float_e  -> mk_constr "Float_e"  []
+        | Float_flag_p, Float_e -> mk_constr "Float_pe" []
+        | Float_flag_s, Float_e -> mk_constr "Float_se" []
+        | Float_flag_, Float_E  -> mk_constr "Float_E"  []
+        | Float_flag_p, Float_E -> mk_constr "Float_pE" []
+        | Float_flag_s, Float_E -> mk_constr "Float_sE" []
+        | Float_flag_, Float_g  -> mk_constr "Float_g"  []
+        | Float_flag_p, Float_g -> mk_constr "Float_pg" []
+        | Float_flag_s, Float_g -> mk_constr "Float_sg" []
+        | Float_flag_, Float_G  -> mk_constr "Float_G"  []
+        | Float_flag_p, Float_G -> mk_constr "Float_pG" []
+        | Float_flag_s, Float_G -> mk_constr "Float_sG" []
+        | Float_flag_, Float_h  -> mk_constr "Float_h"  []
+        | Float_flag_p, Float_h -> mk_constr "Float_ph" []
+        | Float_flag_s, Float_h -> mk_constr "Float_sh" []
+        | Float_flag_, Float_H  -> mk_constr "Float_H"  []
+        | Float_flag_p, Float_H -> mk_constr "Float_pH" []
+        | Float_flag_s, Float_H -> mk_constr "Float_sH" []
+        | Float_flag_, Float_F  -> mk_constr "Float_F"  []
+	| Float_flag_p, Float_F -> mk_constr "Float_pF" []
+	| Float_flag_s, Float_F -> mk_constr "Float_sF" []
       and mk_counter cnt = match cnt with
         | Line_counter  -> mk_constr "Line_counter"  []
         | Char_counter  -> mk_constr "Char_counter"  []
