Index: merlin-4.4-413/src/config/gen_config.ml
===================================================================
--- merlin-4.4-413.orig/src/config/gen_config.ml
+++ merlin-4.4-413/src/config/gen_config.ml
@@ -16,5 +16,5 @@ let ocamlversion :
   [ `OCaml_4_02_0 | `OCaml_4_02_1 | `OCaml_4_02_2 | `OCaml_4_02_3
   | `OCaml_4_03_0 | `OCaml_4_04_0 | `OCaml_4_05_0 | `OCaml_4_06_0
   | `OCaml_4_07_0 | `OCaml_4_07_1 | `OCaml_4_08_0 | `OCaml_4_09_0
-  | `OCaml_4_10_0 | `OCaml_4_11_0 | `OCaml_4_12_0 | `OCaml_4_13_0 ] = %s
+  | `OCaml_4_10_0 | `OCaml_4_11_0 | `OCaml_4_12_0 | `OCaml_4_13_0 | `OCaml_4_14_0 ] = %s
 |} ocaml_version_val
