Index: merlin-3.3.3/src/config/gen_config.ml
===================================================================
--- merlin-3.3.3.orig/src/config/gen_config.ml
+++ merlin-3.3.3/src/config/gen_config.ml
@@ -15,5 +15,5 @@ let version = "%%VERSION%%"
 let ocamlversion :
   [ `OCaml_4_02_0 | `OCaml_4_02_1 | `OCaml_4_02_2 | `OCaml_4_02_3
   | `OCaml_4_03_0 | `OCaml_4_04_0 | `OCaml_4_05_0 | `OCaml_4_06_0
-  | `OCaml_4_07_0 | `OCaml_4_07_1 | `OCaml_4_08_0 | `OCaml_4_09_0 ] = %s
+  | `OCaml_4_07_0 | `OCaml_4_07_1 | `OCaml_4_08_0 | `OCaml_4_09_0 | `OCaml_4_10_0 ] = %s
 |} ocaml_version_val
Index: merlin-3.3.3/src/ocaml/typing/410/typecore.ml
===================================================================
--- merlin-3.3.3.orig/src/ocaml/typing/410/typecore.ml
+++ merlin-3.3.3/src/ocaml/typing/410/typecore.ml
@@ -3800,7 +3800,8 @@ and type_format loc str env =
         | Float_G  -> mk_constr "Float_G"  []
         | Float_h  -> mk_constr "Float_h"  []
         | Float_H  -> mk_constr "Float_H"  []
-        | Float_F  -> mk_constr "Float_F"  [] in
+        | Float_F  -> mk_constr "Float_F"  []
+        | Float_CF -> mk_constr "Float_CF" [] in
         mk_exp_loc (Pexp_tuple [flag; kind])
       and mk_counter cnt = match cnt with
         | Line_counter  -> mk_constr "Line_counter"  []
