Index: async_ssl-0.14.0/bindings/dune
===================================================================
--- async_ssl-0.14.0.orig/bindings/dune
+++ async_ssl-0.14.0/bindings/dune
@@ -16,6 +16,6 @@
   (:standard (:include openssl-ccopt.sexp) \ -Werror -pedantic -Wall
    -Wunused))
  (c_library_flags :standard (:include openssl-cclib.sexp))
- (libraries ctypes.stubs ctypes ctypes.foreign.threaded base)
+ (libraries ctypes.stubs ctypes ctypes.foreign base)
  (virtual_deps conf-openssl) (preprocessor_deps config.h)
- (preprocess (pps ppx_jane)))
\ No newline at end of file
+ (preprocess (pps ppx_jane)))
