Index: eliom-6.9.3/src/lib/eliom_comet_base.shared.ml
===================================================================
--- eliom-6.9.3.orig/src/lib/eliom_comet_base.shared.ml
+++ eliom-6.9.3/src/lib/eliom_comet_base.shared.ml
@@ -49,7 +49,7 @@ type comet_request =
 [@@deriving json]
 
 let comet_request_param =
-  Eliom_parameter.ocaml "comet_request" [%derive.json: comet_request]
+  Eliom_parameter.ocaml "comet_request" [%json: comet_request]
 
 type 'a channel_data =
   | Data of 'a
Index: eliom-6.9.3/src/lib/eliom_common.server.ml
===================================================================
--- eliom-6.9.3.orig/src/lib/eliom_common.server.ml
+++ eliom-6.9.3/src/lib/eliom_common.server.ml
@@ -932,7 +932,7 @@ let get_session_info req previous_extens
           let (tc, pp) =
             List.assoc_remove tab_cookies_param_name post_params
           in
-          let tc = [%derive.of_json: (string * string) list] tc in
+          let tc = [%of_json: (string * string) list] tc in
           (List.fold_left (fun t (k,v) -> CookiesTable.add k v t) CookiesTable.empty tc, pp)
           (*Marshal.from_string (Ocsigen_lib.decode tc) 0, pp*)
         with Not_found ->
@@ -940,7 +940,7 @@ let get_session_info req previous_extens
             let tc = Ocsigen_headers.find tab_cookies_header_name
               (Ocsigen_extensions.Ocsigen_request_info.http_frame ri)
             in
-            let tc = [%derive.of_json: (string * string) list] tc in
+            let tc = [%of_json: (string * string) list] tc in
             (List.fold_left (fun t (k,v) -> CookiesTable.add k v t) CookiesTable.empty tc,
              post_params)
           with Not_found -> CookiesTable.empty, post_params
@@ -955,7 +955,7 @@ let get_session_info req previous_extens
         Ocsigen_headers.find
           tab_cpi_header_name
           (Ocsigen_extensions.Ocsigen_request_info.http_frame ri) in
-      Some ([%derive.of_json: cpi] cpi)
+      Some ([%of_json: cpi] cpi)
     with Not_found -> None
   in
 
@@ -965,7 +965,7 @@ let get_session_info req previous_extens
               expecting_process_page_name
               (Ocsigen_extensions.Ocsigen_request_info.http_frame ri)
             in
-            [%derive.of_json: bool] epd
+            [%of_json: bool] epd
       with Not_found -> false)
   in
 
@@ -1023,7 +1023,7 @@ let get_session_info req previous_extens
     try (* Cookie substitutes for iOS WKWebView *)
       let tc = Ocsigen_headers.find cookie_substitutes_header_name
           (Ocsigen_extensions.Ocsigen_request_info.http_frame ri) in
-      let tc = [%derive.of_json: (string * string) list] tc in
+      let tc = [%of_json: (string * string) list] tc in
         List.fold_left (fun t (k,v) -> CookiesTable.add k v t)
           CookiesTable.empty tc
     with Not_found ->
Index: eliom-6.9.3/src/lib/eliom_cookies_base.shared.ml
===================================================================
--- eliom-6.9.3.orig/src/lib/eliom_cookies_base.shared.ml
+++ eliom-6.9.3/src/lib/eliom_cookies_base.shared.ml
@@ -18,10 +18,10 @@ let cookieset_to_json set =
   in
   let add key v l = (Array.of_list key, cookietable_array v)::l in
   let a = Array.of_list (Cookies.fold add set []) in
-  Deriving_Json.to_string [%derive.json: cookie_array] a
+  Deriving_Json.to_string [%json: cookie_array] a
 
 let cookieset_of_json json =
-  let array = Deriving_Json.from_string [%derive.json: cookie_array] json in
+  let array = Deriving_Json.from_string [%json: cookie_array] json in
   let cookietable_array array =
     Array.fold_left (fun set (name, cookie) -> CookiesTable.add name cookie set)
       CookiesTable.empty array
Index: eliom-6.9.3/src/lib/eliom_runtime.shared.ml
===================================================================
--- eliom-6.9.3.orig/src/lib/eliom_runtime.shared.ml
+++ eliom-6.9.3/src/lib/eliom_runtime.shared.ml
@@ -183,7 +183,7 @@ module RawXML = struct
               match cookie_info with
               | None -> acc_attr
               | Some v ->
-                (ce_call_service_attrib, RA (AStr ([%derive.to_json: cookie_info] v)))
+                (ce_call_service_attrib, RA (AStr ([%json_of: cookie_info] v)))
                 :: acc_attr
             in
             let acc_attr =
