commit 654c44d6b0ee55aca1c205a09cf9ae2eec7f6bb4
Author: Vasilis Papavasileiou <git@vasilis.airpost.net>
Date:   Wed Apr 19 21:32:14 2017 +0200

    JSOO Keycode module no longer exists

diff --git a/src/lib/eliom_content_.client.ml b/src/lib/eliom_content_.client.ml
index f90baecc..c00d9410 100644
--- a/src/lib/eliom_content_.client.ml
+++ b/src/lib/eliom_content_.client.ml
@@ -573,8 +573,7 @@ module Html = struct
         elt##.onscroll := (bool_cb f)
       let onreturn elt f =
         let f ev =
-          let key = ev##.keyCode in
-          if key = Keycode.return then f ev;
+          if Dom_html.Keyboard_code.(of_event ev = Enter) then f ev;
           true in
         onkeydown elt f
       let onchange elt f =
