Index: eliom-11.1.0/src/ppx/ppx_eliom_utils.ml
===================================================================
--- eliom-11.1.0.orig/src/ppx/ppx_eliom_utils.ml
+++ eliom-11.1.0/src/ppx/ppx_eliom_utils.ml
@@ -454,7 +454,7 @@ module Cmo = struct
     type_of_out_type ty
 
   let typ ty =
-    let ty = Printtyp.tree_of_type_scheme ty in
+    let ty = Out_type.tree_of_typexp Out_type.Type_scheme ty in
     type_of_out_type ty
 
   let find err loc =
