Index: eliom-8.6.0/opam
===================================================================
--- eliom-8.6.0.orig/opam
+++ eliom-8.6.0/opam
@@ -14,7 +14,6 @@ depends: [
   "ocaml" {>= "4.08.0"}
   "ocamlfind"
   "ppx_deriving"
-  "ppx_tools" {>= "5.2"}
   "js_of_ocaml-compiler" {>= "3.6.0"}
   "js_of_ocaml" {>= "3.6.0"}
   "js_of_ocaml-lwt" {>= "3.6.0"}
@@ -30,5 +29,5 @@ depends: [
   "reactiveData" {>= "0.2.1"}
   "dbm" | "sqlite3"
   "base-bytes"
-  "ppx_tools_versioned" {>= "5.2.3"}
+  "ppxlib"
 ]
