Index: eliom-10.1.0/src/lib/eliom_shared_content.eliom
===================================================================
--- eliom-10.1.0.orig/src/lib/eliom_shared_content.eliom
+++ eliom-10.1.0/src/lib/eliom_shared_content.eliom
@@ -240,6 +240,11 @@ module Svg = struct
         : (Svg_types.paint -> string) Eliom_shared.Value.t
         :> ([< Svg_types.paint] -> string) Eliom_shared.Value.t)
 
+    let string_of_fill_rule () =
+      ([%shared Raw_wrapped_functions_svg.string_of_fill_rule]
+       : (Svg_types.fill_rule -> string) Eliom_shared.Value.t
+       :> ([< Svg_types.fill_rule] -> string) Eliom_shared.Value.t)
+
     let string_of_strokedasharray () =
       [%shared Raw_wrapped_functions_svg.string_of_strokedasharray]
 
@@ -248,6 +253,7 @@ module Svg = struct
 
     let string_of_transforms () =
       [%shared Raw_wrapped_functions_svg.string_of_transforms]
+
   end
 
   module Xml = struct
@@ -329,6 +335,9 @@ module Html = struct
        : (Html_types.script_type -> string) Eliom_shared.Value.t
        :> ([< Html_types.script_type] -> string) Eliom_shared.Value.t)
 
+    let string_of_autocomplete () =
+      [%shared Raw_wrapped_functions.string_of_autocomplete]
+
     let string_of_linktypes () =
       ([%shared Raw_wrapped_functions.string_of_linktypes]
         : (Html_types.linktype list -> string) Eliom_shared.Value.t
