Index: ocsigen-start-2.13.0/src/os_tips.eliom
===================================================================
--- ocsigen-start-2.13.0.orig/src/os_tips.eliom
+++ ocsigen-start-2.13.0/src/os_tips.eliom
@@ -108,7 +108,7 @@ let%client unset_tip_seen name =
   tips_seen_client_ref := Stringset.remove name !tips_seen_client_ref;
   ~%(Eliom_client.server_function
        ~name:"Os_tips.unset_tip_seen"
-       [%derive.json: string]
+       [%json: string]
        (Os_session.connected_wrapper set_tip_seen))
     name
 
