Index: ocsigen-start-1.1.0/Makefile.options
===================================================================
--- ocsigen-start-1.1.0.orig/Makefile.options
+++ ocsigen-start-1.1.0/Makefile.options
@@ -54,13 +54,13 @@ SASS_TEMPORARY_PROJECT_NAME := os_tempor
 ##----------------------------------------------------------------------
 
 # OCamlfind packages for the server
-SERVER_PACKAGES    := lwt.ppx js_of_ocaml-ppx.deriving calendar safepass \
+SERVER_PACKAGES    := lwt_ppx js_of_ocaml-ppx.deriving calendar safepass \
                       ocsigen-toolkit.server yojson re.str
 
 SERVER_DB_PACKAGES := pgocaml pgocaml.syntax macaque.syntax calendar safepass
 
 # OCamlfind packages for the client
-CLIENT_PACKAGES    := lwt.ppx js_of_ocaml-ppx.deriving js_of_ocaml-ppx \
+CLIENT_PACKAGES    := lwt_ppx js_of_ocaml-ppx.deriving js_of_ocaml-ppx \
                       calendar ocsigen-toolkit.client re.str
 
 # Debug package (yes/no): Debugging info in compilation
Index: ocsigen-start-1.1.0/template.distillery/Makefile.mobile
===================================================================
--- ocsigen-start-1.1.0.orig/template.distillery/Makefile.mobile
+++ ocsigen-start-1.1.0/template.distillery/Makefile.mobile
@@ -232,7 +232,7 @@ $(CORDOVAPATH)/www/eliom.html: $(CORDOVA
 # Eliom loader is used in the index.html to retrieve update from the server.
 mobile/eliom_loader.byte: mobile/eliom_loader.ml
 	ocamlfind ocamlc \
-	    -package js_of_ocaml,js_of_ocaml-ppx,js_of_ocaml-lwt,lwt.ppx \
+	    -package js_of_ocaml,js_of_ocaml-ppx,js_of_ocaml-lwt,lwt_ppx \
 	    -linkpkg -o mobile/eliom_loader.byte \
 	    $<
 
Index: ocsigen-start-1.1.0/template.distillery/Makefile.options
===================================================================
--- ocsigen-start-1.1.0.orig/template.distillery/Makefile.options
+++ ocsigen-start-1.1.0/template.distillery/Makefile.options
@@ -24,13 +24,13 @@ CLIENT_FILES          := $(wildcard *.el
 SERVER_ELIOM_PACKAGES := ocsigen-start.server
 
 # OCamlfind packages for the server
-SERVER_PACKAGES       := lwt.ppx js_of_ocaml-ppx.deriving ppx_deriving.std
+SERVER_PACKAGES       := lwt_ppx js_of_ocaml-ppx.deriving ppx_deriving.std
 
 # OCamlfind packages for database access
 SERVER_DB_PACKAGES    := pgocaml.syntax
 
 # OCamlfind packages for the client
-CLIENT_PACKAGES       := lwt.ppx js_of_ocaml-ppx js_of_ocaml-ppx.deriving \
+CLIENT_PACKAGES       := lwt_ppx js_of_ocaml-ppx js_of_ocaml-ppx.deriving \
                          ppx_deriving.std ocsigen-start.client base
 
 # Automatically install packages via NPM
