Index: ocsigen-start-1.0.0/src/os_fcm_notif.eliom
===================================================================
--- ocsigen-start-1.0.0.orig/src/os_fcm_notif.eliom
+++ ocsigen-start-1.0.0/src/os_fcm_notif.eliom
@@ -400,7 +400,7 @@ module Response =
                content *)
             | None -> Lwt.fail FCM_empty_response
             (* FIXME: is it enough to read a string of length 16384? *)
-            | Some x -> Os_lib.Http.string_of_stream x
+            | Some x -> let%lwt r = Os_lib.Http.bytes_of_stream x in Lwt.return (Bytes.to_string r)
           in
           let content_basic_json =
             Yojson.Safe.to_basic (Yojson.Safe.from_string content_str)
Index: ocsigen-start-1.0.0/src/os_lib.eliom
===================================================================
--- ocsigen-start-1.0.0.orig/src/os_lib.eliom
+++ ocsigen-start-1.0.0/src/os_lib.eliom
@@ -34,10 +34,10 @@ let%shared memoizator f =
 [%%server.start]
 module Http =
   struct
-    let string_of_stream ?(len=16384) contents =
+    let bytes_of_stream ?(len=16384) contents =
       Lwt.try_bind
         (fun () ->
-           Ocsigen_stream.string_of_stream len (Ocsigen_stream.get contents))
+           Ocsigen_stream.bytes_of_stream len (Ocsigen_stream.get contents))
         (fun r ->
            let%lwt () = Ocsigen_stream.finalize contents `Success in
            Lwt.return r)
Index: ocsigen-start-1.0.0/src/os_lib.eliomi
===================================================================
--- ocsigen-start-1.0.0.orig/src/os_lib.eliomi
+++ ocsigen-start-1.0.0/src/os_lib.eliomi
@@ -38,8 +38,8 @@ val memoizator :
 (** This module contains functions about HTTP request. *)
 module Http :
   sig
-    (** [string_of_stream ?len stream] creates a string of maximum length [len]
+    (** [bytes_of_stream ?len stream] creates a string of maximum length [len]
         (default is [16384]) from the stream [stream].
      *)
-    val string_of_stream : ?len:int -> string Ocsigen_stream.t -> string Lwt.t
+    val bytes_of_stream : ?len:int -> bytes Ocsigen_stream.t -> bytes Lwt.t
   end
