Index: ocsigen-start-6.1.0/src/os_user_view.eliom
===================================================================
--- ocsigen-start-6.1.0.orig/src/os_user_view.eliom
+++ ocsigen-start-6.1.0/src/os_user_view.eliom
@@ -229,7 +229,6 @@ let%shared password_form ?(a_placeholder
         D.Form.input
           ~a:
             [ a_required ()
-            ; a_autocomplete false
             ; a_placeholder a_placeholder_pwd ]
           ~input_type:`Password ~name:pwdn D.Form.string
       in
@@ -237,7 +236,6 @@ let%shared password_form ?(a_placeholder
         D.Form.input
           ~a:
             [ a_required ()
-            ; a_autocomplete false
             ; a_placeholder a_placeholder_confirmation ]
           ~input_type:`Password ~name:pwd2n D.Form.string
       in
