Index: pgocaml-3.2/_tags
===================================================================
--- pgocaml-3.2.orig/_tags
+++ pgocaml-3.2/_tags
@@ -109,7 +109,7 @@ true: annot, bin_annot
 "tests/test_re.byte": pkg_calendar
 "tests/test_re.byte": pkg_csv
 "tests/test_re.byte": pkg_hex
-"tests/test_re.byte": pkg_oUnit
+"tests/test_re.byte": pkg_ounit2
 "tests/test_re.byte": pkg_re
 "tests/test_re.byte": pkg_rresult
 "tests/test_re.byte": pkg_unix
@@ -118,7 +118,7 @@ true: annot, bin_annot
 <tests/*.ml{,i,y}>: pkg_calendar
 <tests/*.ml{,i,y}>: pkg_csv
 <tests/*.ml{,i,y}>: pkg_hex
-<tests/*.ml{,i,y}>: pkg_oUnit
+<tests/*.ml{,i,y}>: pkg_ounit2
 <tests/*.ml{,i,y}>: pkg_re
 <tests/*.ml{,i,y}>: pkg_rresult
 <tests/*.ml{,i,y}>: pkg_unix
@@ -129,7 +129,7 @@ true: annot, bin_annot
 "tests_ppx/test_ppx.byte": pkg_compiler-libs.common
 "tests_ppx/test_ppx.byte": pkg_csv
 "tests_ppx/test_ppx.byte": pkg_hex
-"tests_ppx/test_ppx.byte": pkg_oUnit
+"tests_ppx/test_ppx.byte": pkg_ounit2
 "tests_ppx/test_ppx.byte": pkg_ocaml-migrate-parsetree
 "tests_ppx/test_ppx.byte": pkg_ppx_tools_versioned.metaquot_404
 "tests_ppx/test_ppx.byte": pkg_re
@@ -141,7 +141,7 @@ true: annot, bin_annot
 <tests_ppx/*.ml{,i,y}>: pkg_compiler-libs.common
 <tests_ppx/*.ml{,i,y}>: pkg_csv
 <tests_ppx/*.ml{,i,y}>: pkg_hex
-<tests_ppx/*.ml{,i,y}>: pkg_oUnit
+<tests_ppx/*.ml{,i,y}>: pkg_ounit2
 <tests_ppx/*.ml{,i,y}>: pkg_ocaml-migrate-parsetree
 <tests_ppx/*.ml{,i,y}>: pkg_ppx_tools_versioned.metaquot_404
 <tests_ppx/*.ml{,i,y}>: pkg_re
Index: pgocaml-3.2/setup.ml
===================================================================
--- pgocaml-3.2.orig/setup.ml
+++ pgocaml-3.2/setup.ml
@@ -8237,7 +8237,7 @@ let setup_t =
                       bs_build_depends =
                         [
                            InternalLibrary "pgocaml";
-                           FindlibPackage ("oUnit", None)
+                           FindlibPackage ("ounit2", None)
                         ];
                       bs_build_tools =
                         [ExternalTool "ocamlbuild"; ExternalTool "make"];
@@ -8383,7 +8383,7 @@ let setup_t =
                       bs_build_depends =
                         [
                            InternalLibrary "pgocaml";
-                           FindlibPackage ("oUnit", None)
+                           FindlibPackage ("ounit2", None)
                         ];
                       bs_build_tools =
                         [
