Index: ppx_deriving_protobuf-2.7/src/ppx_deriving_protobuf.cppo.ml
===================================================================
--- ppx_deriving_protobuf-2.7.orig/src/ppx_deriving_protobuf.cppo.ml
+++ ppx_deriving_protobuf-2.7/src/ppx_deriving_protobuf.cppo.ml
@@ -1170,9 +1170,9 @@ let rec write_protoc ~fmt ~path:base_pat
       let i = int_of_string sn in
 #endif
       Format.fprintf fmt " [default=%d]" i
-    | Some { pexp_desc = Pexp_constant (Pconst_string (s, _)) } ->
+    | Some { pexp_desc = Pexp_constant (Pconst_string (s, _, _)) } ->
       Format.fprintf fmt " [default=\"%s\"]" (escape ~pass_8bit:true s)
-    | Some [%expr Bytes.of_string [%e? { pexp_desc = Pexp_constant (Pconst_string (s, _)) }]] ->
+    | Some [%expr Bytes.of_string [%e? { pexp_desc = Pexp_constant (Pconst_string (s, _, _)) }]] ->
       Format.fprintf fmt " [default=\"%s\"]" (escape ~pass_8bit:false s)
     | Some { pexp_desc = Pexp_construct ({ txt = Lident n }, _) }
     | Some { pexp_desc = Pexp_variant (n, _) } ->
