Index: ppx_deriving_protobuf-2.7/src_test/dune
===================================================================
--- ppx_deriving_protobuf-2.7.orig/src_test/dune
+++ ppx_deriving_protobuf-2.7/src_test/dune
@@ -1,6 +1,6 @@
 (executable
  (name test_ppx_protobuf)
- (libraries str uint oUnit)
+ (libraries str uint ounit2)
  (preprocess
   (pps ppx_deriving_protobuf)))
 
Index: ppx_deriving_protobuf-2.7/ppx_deriving_protobuf.opam
===================================================================
--- ppx_deriving_protobuf-2.7.orig/ppx_deriving_protobuf.opam
+++ ppx_deriving_protobuf-2.7/ppx_deriving_protobuf.opam
@@ -20,7 +20,7 @@ depends: [
   "ppx_tools"    {build}
   "cppo"         {build}
   "ppx_deriving" {>= "4.0" & < "5.0"}
-  "ounit"        {with-test}
+  "ounit2"       {with-test}
   "uint"         {with-test}
 ]
 synopsis: "A Protocol Buffers codec generator for OCaml >=4.02"
