Index: ocaml-pcap-0.5.2/lib_test/dune
===================================================================
--- ocaml-pcap-0.5.2.orig/lib_test/dune
+++ ocaml-pcap-0.5.2/lib_test/dune
@@ -1,6 +1,6 @@
 (executable
  (name test)
- (libraries pcap-format oUnit mmap))
+ (libraries pcap-format ounit2 mmap))
 
 (alias
  (name runtest)
Index: ocaml-pcap-0.5.2/pcap-format.opam
===================================================================
--- ocaml-pcap-0.5.2.orig/pcap-format.opam
+++ ocaml-pcap-0.5.2/pcap-format.opam
@@ -12,7 +12,7 @@ depends: [
   "ppx_tools"
   "cstruct" {>= "1.9.0"}
   "ppx_cstruct"
-  "ounit" {with-test}
+  "ounit2" {with-test}
   "mmap" {with-test}
 ]
 build: [
