Index: ocaml-cohttp-1.1.1/cohttp-async/src/client.ml
===================================================================
--- ocaml-cohttp-1.1.1.orig/cohttp-async/src/client.ml
+++ ocaml-cohttp-1.1.1/cohttp-async/src/client.ml
@@ -26,7 +26,7 @@ module Net = struct
       >>| function
       | { Addr_info.ai_addr=ADDR_INET (addr,_); _ }::_ ->
         Or_error.return (host, Ipaddr_unix.of_inet_addr addr, port)
-      | _ -> Or_error.error "Failed to resolve Uri" uri Uri.sexp_of_t
+      | _ -> Or_error.error "Failed to resolve Uri" uri Uri_sexp.sexp_of_t
 
   let connect_uri ?interrupt ?ssl_config uri =
     lookup uri
Index: ocaml-cohttp-1.1.1/cohttp-async/src/dune
===================================================================
--- ocaml-cohttp-1.1.1.orig/cohttp-async/src/dune
+++ ocaml-cohttp-1.1.1/cohttp-async/src/dune
@@ -3,5 +3,5 @@
   (synopsis    "Async backend")
   (public_name cohttp-async)
   (libraries   logs.fmt base fmt async_unix async_kernel async_extra uri
-               uri.services ipaddr.unix conduit-async magic-mime cohttp)
+               uri.services uri.sexp ipaddr.unix conduit-async magic-mime cohttp)
   (preprocess (pps ppx_sexp_conv)))
