Index: reason-3.0.4/src/reason-parser/jbuild
===================================================================
--- reason-3.0.4.orig/src/reason-parser/jbuild
+++ reason-3.0.4/src/reason-parser/jbuild
@@ -10,7 +10,7 @@
  ((targets (reason_parser.messages))
   (deps (reason_parser.mly reason_parser.messages.checked-in))
   (action
-    (with-stdout-to ${@} (bash "menhir --explain --strict --unused-tokens ./reason_parser.mly --update-errors ./reason_parser.messages.checked-in | sed -e 's/[[:space:]]*$$//g'"))
+    (with-stdout-to ${@} (bash "menhir --explain --unused-tokens ./reason_parser.mly --update-errors ./reason_parser.messages.checked-in | sed -e 's/[[:space:]]*$$//g'"))
    )))
 
 ; Previously, make preprocess.
@@ -18,7 +18,7 @@
  ((targets (reason_parser.cmly reason_parser.ml reason_parser.mli))
   (deps (reason_parser.mly reason_parser.messages))
   (action
-   (run menhir --strict --unused-tokens --fixed-exception --table --cmly reason_parser.mly))))
+   (run menhir --unused-tokens --fixed-exception --table --cmly reason_parser.mly))))
 
 (executables
  ((names (menhir_error_processor))
@@ -40,7 +40,7 @@
  ((targets (reason_parser_message.ml))
   (deps (reason_parser.mly reason_parser.messages))
   (action
-   (with-stdout-to ${@} (run menhir --explain --strict --unused-tokens reason_parser.mly --compile-errors reason_parser.messages)))))
+   (with-stdout-to ${@} (run menhir --explain --unused-tokens reason_parser.mly --compile-errors reason_parser.messages)))))
 
 (library
  ((name reason)
