Index: reason-3.0.2/src/reason-parser/reason_lexer.mll
===================================================================
--- reason-3.0.2.orig/src/reason-parser/reason_lexer.mll
+++ reason-3.0.2/src/reason-parser/reason_lexer.mll
@@ -350,7 +350,7 @@ let preprocessor = ref None
 
 let warn_latin1 lexbuf =
   Location.prerr_warning (Location.curr lexbuf)
-    (Warnings.Deprecated "ISO-Latin1 characters in identifiers")
+    (Warnings.Deprecated ("ISO-Latin1 characters in identifiers",(Location.curr lexbuf),(Location.curr lexbuf)))
 ;;
 
 (* Error report *)
