Index: ocaml-ctypes-0.15.1/Makefile.tests
===================================================================
--- ocaml-ctypes-0.15.1.orig/Makefile.tests
+++ ocaml-ctypes-0.15.1/Makefile.tests
@@ -19,7 +19,7 @@ tests-common: $$(LIB_TARGETS)
 
 test-raw.dir = tests/test-raw
 test-raw.threads = yes
-test-raw.deps = bigarray oUnit str bytes integers
+test-raw.deps = bigarray ounit2 str bytes integers
 test-raw.subproject_deps = ctypes ctypes-foreign-base ctypes-foreign-threaded
 test-raw: PROJECT=test-raw
 test-raw: $$(BEST_TARGET)
@@ -41,7 +41,7 @@ test-pointers-stub-generator: $$(BEST_TA
 
 test-pointers.dir = tests/test-pointers
 test-pointers.threads = yes
-test-pointers.deps = str bigarray oUnit bytes integers
+test-pointers.deps = str bigarray ounit2 bytes integers
 test-pointers.subproject_deps = ctypes ctypes-foreign-base \
   ctypes-foreign-threaded cstubs tests-common test-pointers-stubs
 test-pointers.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -76,7 +76,7 @@ test-integers-stub-generator: $$(BEST_TA
 
 test-integers.dir = tests/test-integers
 test-integers.threads = yes
-test-integers.deps = str bigarray oUnit bytes integers
+test-integers.deps = str bigarray ounit2 bytes integers
 test-integers.subproject_deps = ctypes ctypes-foreign-base \
   ctypes-foreign-threaded cstubs tests-common test-integers-stubs
 test-integers.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -112,7 +112,7 @@ test-variadic-stub-generator: $$(BEST_TA
 
 test-variadic.dir = tests/test-variadic
 test-variadic.threads = yes
-test-variadic.deps = str bigarray oUnit bytes integers
+test-variadic.deps = str bigarray ounit2 bytes integers
 test-variadic.subproject_deps = ctypes ctypes-foreign-base \
   ctypes-foreign-threaded cstubs tests-common test-variadic-stubs
 test-variadic.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -147,7 +147,7 @@ test-builtins-stub-generator: $$(BEST_TA
 
 test-builtins.dir = tests/test-builtins
 test-builtins.threads = yes
-test-builtins.deps = str bigarray oUnit bytes integers
+test-builtins.deps = str bigarray ounit2 bytes integers
 test-builtins.subproject_deps = ctypes cstubs test-builtins-stubs \
   ctypes-foreign-base ctypes-foreign-threaded tests-common
 test-builtins.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -182,7 +182,7 @@ test-macros-stub-generator: $$(BEST_TARG
 
 test-macros.dir = tests/test-macros
 test-macros.threads = yes
-test-macros.deps = str bigarray oUnit bytes integers
+test-macros.deps = str bigarray ounit2 bytes integers
 test-macros.subproject_deps = ctypes cstubs test-macros-stubs \
   ctypes-foreign-base ctypes-foreign-threaded tests-common
 test-macros.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -218,7 +218,7 @@ test-higher_order-stub-generator: $$(BES
 
 test-higher_order.dir = tests/test-higher_order
 test-higher_order.threads = yes
-test-higher_order.deps = str bigarray oUnit bytes integers
+test-higher_order.deps = str bigarray ounit2 bytes integers
 test-higher_order.subproject_deps = ctypes ctypes-foreign-base \
   ctypes-foreign-threaded cstubs test-higher_order-stubs tests-common
 test-higher_order.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -272,7 +272,7 @@ test-enums-struct-stub-generator: $$(BES
 
 test-enums.dir = tests/test-enums
 test-enums.threads = yes
-test-enums.deps = str bigarray oUnit bytes integers
+test-enums.deps = str bigarray ounit2 bytes integers
 test-enums.subproject_deps = ctypes ctypes-foreign-base \
   ctypes-foreign-threaded cstubs test-enums-struct-stubs test-enums-stubs tests-common
 test-enums.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -319,7 +319,7 @@ test-structs-stub-generator: $$(BEST_TAR
 
 test-structs.dir = tests/test-structs
 test-structs.threads = yes
-test-structs.deps = str bigarray oUnit bytes integers
+test-structs.deps = str bigarray ounit2 bytes integers
 test-structs.subproject_deps = ctypes ctypes-foreign-base \
   ctypes-foreign-threaded cstubs test-structs-stubs tests-common
 test-structs.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -362,7 +362,7 @@ test-constants-stub-generator: $$(BEST_T
 
 test-constants.dir = tests/test-constants
 test-constants.threads = yes
-test-constants.deps = str bigarray oUnit bytes integers
+test-constants.deps = str bigarray ounit2 bytes integers
 test-constants.subproject_deps = ctypes ctypes-foreign-base \
   ctypes-foreign-threaded cstubs test-constants-stubs tests-common
 test-constants.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -391,7 +391,7 @@ $(BUILDDIR)/tests/test-constants/generat
 
 test-finalisers.dir = tests/test-finalisers
 test-finalisers.threads = yes
-test-finalisers.deps = str bigarray oUnit bytes integers
+test-finalisers.deps = str bigarray ounit2 bytes integers
 test-finalisers.subproject_deps = ctypes ctypes-foreign-base ctypes-foreign-threaded
 test-finalisers: PROJECT=test-finalisers
 test-finalisers: $$(BEST_TARGET)
@@ -413,7 +413,7 @@ test-cstdlib-stub-generator: $$(BEST_TAR
 
 test-cstdlib.dir = tests/test-cstdlib
 test-cstdlib.threads = yes
-test-cstdlib.deps = str bigarray oUnit bytes integers
+test-cstdlib.deps = str bigarray ounit2 bytes integers
 test-cstdlib.subproject_deps = ctypes ctypes-foreign-base \
    ctypes-foreign-threaded cstubs test-cstdlib-stubs tests-common
 test-cstdlib.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -433,7 +433,7 @@ tests/test-cstdlib/generated_bindings.ml
 
 test-sizeof.dir = tests/test-sizeof
 test-sizeof.threads = yes
-test-sizeof.deps = str bigarray oUnit bytes integers
+test-sizeof.deps = str bigarray ounit2 bytes integers
 test-sizeof.subproject_deps = ctypes ctypes-foreign-base ctypes-foreign-threaded
 test-sizeof: PROJECT=test-sizeof
 test-sizeof: $$(BEST_TARGET)
@@ -455,7 +455,7 @@ test-foreign_values-stub-generator: $$(B
 
 test-foreign_values.dir = tests/test-foreign_values
 test-foreign_values.threads = yes
-test-foreign_values.deps = str bigarray oUnit bytes integers
+test-foreign_values.deps = str bigarray ounit2 bytes integers
 test-foreign_values.subproject_deps = ctypes ctypes-foreign-base \
   ctypes-foreign-threaded cstubs tests-common test-foreign_values-stubs
 test-foreign_values.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -490,7 +490,7 @@ test-unions-stub-generator: $$(BEST_TARG
 
 test-unions.dir = tests/test-unions
 test-unions.threads = yes
-test-unions.deps = str bigarray oUnit bytes integers
+test-unions.deps = str bigarray ounit2 bytes integers
 test-unions.subproject_deps = ctypes ctypes-foreign-base \
    ctypes-foreign-threaded cstubs test-unions-stubs tests-common
 test-unions.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -518,7 +518,7 @@ $(BUILDDIR)/tests/test-unions/generated_
 
 test-custom_ops.dir = tests/test-custom_ops
 test-custom_ops.threads = yes
-test-custom_ops.deps = str bigarray oUnit bytes integers
+test-custom_ops.deps = str bigarray ounit2 bytes integers
 test-custom_ops.subproject_deps = ctypes ctypes-foreign-base ctypes-foreign-threaded
 test-custom_ops: PROJECT=test-custom_ops
 test-custom_ops: $$(BEST_TARGET)
@@ -540,7 +540,7 @@ test-arrays-stub-generator: $$(BEST_TARG
 
 test-arrays.dir = tests/test-arrays
 test-arrays.threads = yes
-test-arrays.deps = str bigarray oUnit bytes integers
+test-arrays.deps = str bigarray ounit2 bytes integers
 test-arrays.subproject_deps = ctypes ctypes-foreign-base \
   ctypes-foreign-threaded cstubs test-arrays-stubs tests-common
 test-arrays.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -560,21 +560,21 @@ tests/test-arrays/generated_bindings.ml:
 
 test-foreign-errno.dir = tests/test-foreign-errno
 test-foreign-errno.threads = yes
-test-foreign-errno.deps = str bigarray oUnit bytes integers
+test-foreign-errno.deps = str bigarray ounit2 bytes integers
 test-foreign-errno.subproject_deps = ctypes ctypes-foreign-base ctypes-foreign-threaded
 test-foreign-errno: PROJECT=test-foreign-errno
 test-foreign-errno: $$(BEST_TARGET)
 
 test-passable.dir = tests/test-passable
 test-passable.threads = yes
-test-passable.deps = str bigarray oUnit bytes integers
+test-passable.deps = str bigarray ounit2 bytes integers
 test-passable.subproject_deps = ctypes ctypes-foreign-base ctypes-foreign-threaded cstubs
 test-passable: PROJECT=test-passable
 test-passable: $$(BEST_TARGET)
 
 test-alignment.dir = tests/test-alignment
 test-alignment.threads = yes
-test-alignment.deps = str bigarray oUnit bytes integers
+test-alignment.deps = str bigarray ounit2 bytes integers
 test-alignment.subproject_deps = ctypes ctypes-foreign-base ctypes-foreign-threaded
 test-alignment: PROJECT=test-alignment
 test-alignment: $$(BEST_TARGET)
@@ -596,7 +596,7 @@ test-views-stub-generator: $$(BEST_TARGE
 
 test-views.dir = tests/test-views
 test-views.threads = yes
-test-views.deps = str bigarray oUnit bytes integers
+test-views.deps = str bigarray ounit2 bytes integers
 test-views.subproject_deps = ctypes ctypes-foreign-base ctypes-foreign-threaded cstubs test-views-stubs tests-common
 test-views.link_flags = -L$(BUILDDIR)/clib -ltest_functions
 test-views: PROJECT=test-views
@@ -630,7 +630,7 @@ test-oo_style-stub-generator: $$(BEST_TA
 
 test-oo_style.dir = tests/test-oo_style
 test-oo_style.threads = yes
-test-oo_style.deps = str bigarray oUnit bytes integers
+test-oo_style.deps = str bigarray ounit2 bytes integers
 test-oo_style.subproject_deps = ctypes ctypes-foreign-base \
    ctypes-foreign-threaded cstubs test-oo_style-stubs tests-common
 test-oo_style.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -650,7 +650,7 @@ tests/test-oo_style/generated_bindings.m
 
 test-marshal.dir = tests/test-marshal
 test-marshal.threads = yes
-test-marshal.deps = str bigarray oUnit bytes integers
+test-marshal.deps = str bigarray ounit2 bytes integers
 test-marshal.subproject_deps = ctypes ctypes-foreign-base \
    ctypes-foreign-threaded cstubs tests-common
 test-marshal.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -659,7 +659,7 @@ test-marshal: $$(BEST_TARGET)
 
 test-type_printing.dir = tests/test-type_printing
 test-type_printing.threads = yes
-test-type_printing.deps = str bigarray oUnit bytes integers
+test-type_printing.deps = str bigarray ounit2 bytes integers
 test-type_printing.subproject_deps = ctypes ctypes-foreign-base ctypes-foreign-threaded \
   cstubs test-type_printing-stubs tests-common
 test-type_printing.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -711,7 +711,7 @@ test-value_printing-stub-generator: $$(B
 
 test-value_printing.dir = tests/test-value_printing
 test-value_printing.threads = yes
-test-value_printing.deps = str bigarray oUnit bytes integers
+test-value_printing.deps = str bigarray ounit2 bytes integers
 test-value_printing.subproject_deps = ctypes ctypes-foreign-base \
    ctypes-foreign-threaded cstubs tests-common test-value_printing-stubs
 test-value_printing.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -746,7 +746,7 @@ test-complex-stub-generator: $$(BEST_TAR
 
 test-complex.dir = tests/test-complex
 test-complex.threads = yes
-test-complex.deps = str bigarray oUnit bytes integers
+test-complex.deps = str bigarray ounit2 bytes integers
 test-complex.subproject_deps = ctypes ctypes-foreign-base \
   ctypes-foreign-threaded cstubs tests-common test-complex-stubs
 test-complex.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -781,7 +781,7 @@ test-bools-stub-generator: $$(BEST_TARGE
 
 test-bools.dir = tests/test-bools
 test-bools.threads = yes
-test-bools.deps = str bigarray oUnit bytes integers
+test-bools.deps = str bigarray ounit2 bytes integers
 test-bools.subproject_deps = ctypes ctypes-foreign-base \
   ctypes-foreign-threaded cstubs tests-common test-bools-stubs
 test-bools.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -816,7 +816,7 @@ test-callback_lifetime-stub-generator: $
 
 test-callback_lifetime.dir = tests/test-callback_lifetime
 test-callback_lifetime.threads = yes
-test-callback_lifetime.deps = str bigarray oUnit bytes integers
+test-callback_lifetime.deps = str bigarray ounit2 bytes integers
 test-callback_lifetime.subproject_deps = ctypes ctypes-foreign-base \
   ctypes-foreign-threaded cstubs test-callback_lifetime-stubs tests-common
 test-callback_lifetime.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -851,7 +851,7 @@ test-lifetime-stub-generator: $$(BEST_TA
 
 test-lifetime.dir = tests/test-lifetime
 test-lifetime.threads = yes
-test-lifetime.deps = str bigarray oUnit bytes integers
+test-lifetime.deps = str bigarray ounit2 bytes integers
 test-lifetime.subproject_deps = ctypes ctypes-foreign-base \
   ctypes-foreign-threaded cstubs test-lifetime-stubs tests-common
 test-lifetime.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -871,7 +871,7 @@ tests/test-lifetime/generated_bindings.m
 
 test-stubs.dir = tests/test-stubs
 test-stubs.threads = yes
-test-stubs.deps = str bigarray oUnit bytes integers
+test-stubs.deps = str bigarray ounit2 bytes integers
 test-stubs.subproject_deps = ctypes ctypes-foreign-base ctypes-foreign-threaded
 test-stubs: PROJECT=test-stubs
 test-stubs: $$(BEST_TARGET)
@@ -893,7 +893,7 @@ test-bigarrays-stub-generator: $$(BEST_T
 
 test-bigarrays.dir = tests/test-bigarrays
 test-bigarrays.threads = yes
-test-bigarrays.deps = str bigarray oUnit bytes integers
+test-bigarrays.deps = str bigarray ounit2 bytes integers
 test-bigarrays.subproject_deps = ctypes ctypes-foreign-base \
    ctypes-foreign-threaded cstubs tests-common test-bigarrays-stubs
 test-bigarrays.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -928,7 +928,7 @@ test-coercions-stub-generator: $$(BEST_T
 
 test-coercions.dir = tests/test-coercions
 test-coercions.threads = yes
-test-coercions.deps = str bigarray oUnit bytes integers
+test-coercions.deps = str bigarray ounit2 bytes integers
 test-coercions.subproject_deps = ctypes ctypes-foreign-base \
    ctypes-foreign-threaded cstubs tests-common test-coercions-stubs
 test-coercions.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -948,7 +948,7 @@ tests/test-coercions/generated_bindings.
 
 test-roots.dir = tests/test-roots
 test-roots.threads = yes
-test-roots.deps = str bigarray oUnit bytes integers
+test-roots.deps = str bigarray ounit2 bytes integers
 test-roots.subproject_deps = ctypes ctypes-foreign-base \
    ctypes-foreign-threaded cstubs tests-common
 test-roots.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -972,7 +972,7 @@ test-passing-ocaml-values-stub-generator
 
 test-passing-ocaml-values.dir = tests/test-passing-ocaml-values
 test-passing-ocaml-values.threads = yes
-test-passing-ocaml-values.deps = str bigarray oUnit bytes integers
+test-passing-ocaml-values.deps = str bigarray ounit2 bytes integers
 test-passing-ocaml-values.subproject_deps = ctypes ctypes-foreign-base \
    ctypes-foreign-threaded cstubs tests-common test-passing-ocaml-values-stubs
 test-passing-ocaml-values.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -1007,7 +1007,7 @@ test-lwt-jobs-stub-generator: $$(BEST_TA
 
 test-lwt-jobs.dir = tests/test-lwt-jobs
 test-lwt-jobs.threads = yes
-test-lwt-jobs.deps = str bigarray oUnit bytes integers lwt.unix
+test-lwt-jobs.deps = str bigarray ounit2 bytes integers lwt.unix
 test-lwt-jobs.subproject_deps = ctypes ctypes-foreign-base \
    ctypes-foreign-threaded cstubs tests-common test-lwt-jobs-stubs
 test-lwt-jobs.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -1050,7 +1050,7 @@ test-lwt-preemptive-stub-generator: $$(B
 
 test-lwt-preemptive.dir = tests/test-lwt-preemptive
 test-lwt-preemptive.threads = yes
-test-lwt-preemptive.deps = str bigarray oUnit bytes integers lwt.unix
+test-lwt-preemptive.deps = str bigarray ounit2 bytes integers lwt.unix
 test-lwt-preemptive.subproject_deps = ctypes ctypes-foreign-base \
    ctypes-foreign-threaded cstubs tests-common test-lwt-preemptive-stubs
 test-lwt-preemptive.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -1093,7 +1093,7 @@ test-returning-errno-lwt-jobs-stub-gener
 
 test-returning-errno-lwt-jobs.dir = tests/test-returning-errno-lwt-jobs
 test-returning-errno-lwt-jobs.threads = yes
-test-returning-errno-lwt-jobs.deps = str bigarray oUnit bytes integers lwt.unix
+test-returning-errno-lwt-jobs.deps = str bigarray ounit2 bytes integers lwt.unix
 test-returning-errno-lwt-jobs.subproject_deps = ctypes ctypes-foreign-base \
    ctypes-foreign-threaded cstubs tests-common test-returning-errno-lwt-jobs-stubs
 test-returning-errno-lwt-jobs.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -1135,7 +1135,7 @@ test-returning-errno-lwt-preemptive-stub
 
 test-returning-errno-lwt-preemptive.dir = tests/test-returning-errno-lwt-preemptive
 test-returning-errno-lwt-preemptive.threads = yes
-test-returning-errno-lwt-preemptive.deps = str bigarray oUnit bytes integers lwt.unix
+test-returning-errno-lwt-preemptive.deps = str bigarray ounit2 bytes integers lwt.unix
 test-returning-errno-lwt-preemptive.subproject_deps = ctypes ctypes-foreign-base \
    ctypes-foreign-threaded cstubs tests-common test-returning-errno-lwt-preemptive-stubs
 test-returning-errno-lwt-preemptive.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -1177,7 +1177,7 @@ test-returning-errno-stub-generator: $$(
 
 test-returning-errno.dir = tests/test-returning-errno
 test-returning-errno.threads = yes
-test-returning-errno.deps = str bigarray oUnit bytes integers
+test-returning-errno.deps = str bigarray ounit2 bytes integers
 test-returning-errno.subproject_deps = ctypes ctypes-foreign-base \
    ctypes-foreign-threaded cstubs tests-common test-returning-errno-stubs
 test-returning-errno.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -1220,7 +1220,7 @@ test-threads-stub-generator: $$(BEST_TAR
 
 test-threads.dir = tests/test-threads
 test-threads.threads = yes
-test-threads.deps = str bigarray oUnit bytes integers
+test-threads.deps = str bigarray ounit2 bytes integers
 test-threads.subproject_deps = ctypes ctypes-foreign-base \
    ctypes-foreign-threaded cstubs tests-common test-threads-stubs
 test-threads.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -1255,7 +1255,7 @@ test-closure-type-promotion-stub-generat
 
 test-closure-type-promotion.dir = tests/test-closure-type-promotion
 test-closure-type-promotion.threads = yes
-test-closure-type-promotion.deps = str bigarray oUnit bytes integers
+test-closure-type-promotion.deps = str bigarray ounit2 bytes integers
 test-closure-type-promotion.subproject_deps = ctypes ctypes-foreign-base \
   ctypes-foreign-threaded cstubs test-closure-type-promotion-stubs tests-common
 test-closure-type-promotion.link_flags = -L$(BUILDDIR)/clib -ltest_functions
@@ -1275,7 +1275,7 @@ tests/test-closure-type-promotion/genera
 
 test-ldouble.dir = tests/test-ldouble
 test-ldouble.threads = yes
-test-ldouble.deps = str bigarray oUnit bytes integers
+test-ldouble.deps = str bigarray ounit2 bytes integers
 test-ldouble.subproject_deps = ctypes 
 test-ldouble: PROJECT=test-ldouble
 test-ldouble: $$(BEST_TARGET)
