Index: labltk-8.06.9/browser/typecheck.ml
===================================================================
--- labltk-8.06.9.orig/browser/typecheck.ml
+++ labltk-8.06.9/browser/typecheck.ml
@@ -117,7 +117,7 @@ let f txt =
     begin function
       Ptop_def pstr ->
         let str, sign, _names, env' =
-          Typemod.type_structure !env pstr Location.none in
+          Typemod.type_structure !env pstr in
         txt.structure <- txt.structure @ str.str_items;
         txt.signature <- txt.signature @ sign;
         env := env'
