Index: OCaml-ImageMagick-0.34/magick.ml
===================================================================
--- OCaml-ImageMagick-0.34.orig/magick.ml
+++ OCaml-ImageMagick-0.34/magick.ml
@@ -912,9 +912,9 @@ let color_of_hex str_color =
 
   (* convert a char to a string *)
   let string_of_char my_char =
-    let str = " " in
+    let str = (Bytes.of_string " ") in
     str.[0] <- my_char;
-    str
+    Bytes.to_string str
   in
 
   (* cast and check the values of the color for IM *)
