Index: ppx_js_style-0.17.0/src/ppx_js_style.ml
===================================================================
--- ppx_js_style-0.17.0.orig/src/ppx_js_style.ml
+++ ppx_js_style-0.17.0/src/ppx_js_style.ml
@@ -121,7 +121,6 @@ let ignored_expr_must_be_annotated ignor
   (* no need to warn people trying to silence other warnings *)
   | Pexp_construct _
   | Pexp_ident _
-  | Pexp_fun _
   | Pexp_function _ -> ()
   | _ -> f ~loc:expr.pexp_loc (Missing_type_annotation ignored_reason)
 ;;
