Index: js_of_ocaml-3.1.0/lib/lwt/log/jbuild
===================================================================
--- js_of_ocaml-3.1.0.orig/lib/lwt/log/jbuild
+++ js_of_ocaml-3.1.0/lib/lwt/log/jbuild
@@ -3,6 +3,5 @@
  ((name lwt_log_js)
   (public_name js_of_ocaml-lwt.logger)
   (synopsis "Lwt logger for js_of_ocaml.")
-  (optional)
-  (libraries (js_of_ocaml lwt lwt.log))
+  (libraries (js_of_ocaml lwt lwt_log))
   (preprocess (pps (js_of_ocaml-ppx)))))
