Index: lambdasoup-0.7.3/lambdasoup.opam
===================================================================
--- lambdasoup-0.7.3.orig/lambdasoup.opam
+++ lambdasoup-0.7.3/lambdasoup.opam
@@ -16,6 +16,7 @@ depends: [
   "dune" {>= "2.7.0"}
   "markup" {>= "1.0.0"}
   "ocaml" {>= "4.02.0"}
+  "camlp-streams"
 
   "bisect_ppx" {dev & >= "2.5.0"}
   "ounit2" {with-test}
Index: lambdasoup-0.7.3/src/dune
===================================================================
--- lambdasoup-0.7.3.orig/src/dune
+++ lambdasoup-0.7.3/src/dune
@@ -5,4 +5,4 @@
  (wrapped false)
  (flags (:standard -w +A -warn-error -3))
  (instrumentation (backend bisect_ppx))
- (libraries markup))
+ (libraries markup camlp-streams))
Index: lambdasoup-0.7.3/src/soup.ml
===================================================================
--- lambdasoup-0.7.3.orig/src/soup.ml
+++ lambdasoup-0.7.3/src/soup.ml
@@ -27,19 +27,6 @@ struct
       in
       let suffix_length = measure_suffix 0 in
       String.sub s 0 (String.length s - suffix_length)
-
-  (* Convert multiple deprecation warnings into one warning. This function can
-     be removed from the module if/when Lambda Soup supports only
-     OCaml >= 4.03. *)
-  let lowercase_ascii = lowercase [@ocaml.warning "-3"]
-end
-
-module Char =
-struct
-  include Char
-
-  (* See comment by String.lowercase_ascii above. *)
-  let lowercase_ascii = lowercase [@ocaml.warning "-3"]
 end
 
 type element = unit
