Index: async_js-0.12.0/src/dune
===================================================================
--- async_js-0.12.0.orig/src/dune
+++ async_js-0.12.0/src/dune
@@ -1,3 +1,3 @@
 (library (name async_js) (public_name async_js)
- (libraries js_of_ocaml async_kernel async_rpc_kernel uri uri.sexp)
- (preprocess (pps ppx_jane js_of_ocaml-ppx)))
\ No newline at end of file
+ (libraries js_of_ocaml async_kernel async_rpc_kernel uri uri-sexp)
+ (preprocess (pps ppx_jane js_of_ocaml-ppx)))
Index: async_js-0.12.0/src/ocaml_uri.ml
===================================================================
--- async_js-0.12.0.orig/src/ocaml_uri.ml
+++ async_js-0.12.0/src/ocaml_uri.ml
@@ -1,3 +1,2 @@
 module Uri      = Uri
-module Uri_re   = Uri_re
 module Uri_sexp = Uri_sexp
