Index: async_js-0.11.0/src/jbuild
===================================================================
--- async_js-0.11.0.orig/src/jbuild
+++ async_js-0.11.0/src/jbuild
@@ -6,6 +6,7 @@
               async_kernel
               async_rpc_kernel
               uri
+	      uri.sexp
               ))
   (preprocess (pps (ppx_jane js_of_ocaml-ppx ppxlib.runner)))))
 
Index: async_js-0.11.0/src/rpc.ml
===================================================================
--- async_js-0.11.0.orig/src/rpc.ml
+++ async_js-0.11.0/src/rpc.ml
@@ -148,8 +148,8 @@ module Connection = struct
     in
     let pipe_reader,pipe_writer = pipe_of_websocket uri in
     let description = match description with
-      | None -> Info.create "Client connected via WS" uri Uri.sexp_of_t
-      | Some desc -> Info.tag_arg desc "via WS" uri Uri.sexp_of_t
+      | None -> Info.create "Client connected via WS" uri Uri_sexp.sexp_of_t
+      | Some desc -> Info.tag_arg desc "via WS" uri Uri_sexp.sexp_of_t
     in
     let transport = Pipe_transport.create Pipe_transport.Kind.bigstring pipe_reader pipe_writer in
     begin
