Index: ocaml-data-notation-0.0.11/_tags
===================================================================
--- ocaml-data-notation-0.0.11.orig/_tags
+++ ocaml-data-notation-0.0.11/_tags
@@ -46,11 +46,11 @@ true: annot, bin_annot
 <src/*.ml{,i,y}>: pkg_type_conv
 # Executable test
 "test/test.byte": pkg_fileutils
-"test/test.byte": pkg_oUnit
+"test/test.byte": pkg_ounit2
 "test/test.byte": pkg_str
 "test/test.byte": use_odn
 <test/*.ml{,i,y}>: pkg_fileutils
-<test/*.ml{,i,y}>: pkg_oUnit
+<test/*.ml{,i,y}>: pkg_ounit2
 <test/*.ml{,i,y}>: pkg_str
 <test/*.ml{,i,y}>: use_odn
 # OASIS_STOP
Index: ocaml-data-notation-0.0.11/setup.ml
===================================================================
--- ocaml-data-notation-0.0.11.orig/setup.ml
+++ ocaml-data-notation-0.0.11/setup.ml
@@ -7525,7 +7525,7 @@ let setup_t =
                       bs_build_depends =
                         [
                            FindlibPackage
-                             ("oUnit",
+                             ("ounit2",
                                Some (OASISVersion.VGreaterEqual "2.0.0"));
                            FindlibPackage
                              ("fileutils",
