Index: ocaml-expect-0.0.6/_tags
===================================================================
--- ocaml-expect-0.0.6.orig/_tags
+++ ocaml-expect-0.0.6/_tags
@@ -29,7 +29,7 @@ true: annot, bin_annot
 # Executable test
 "test/test.byte": pkg_batteries
 "test/test.byte": pkg_bytes
-"test/test.byte": pkg_oUnit
+"test/test.byte": pkg_ounit2
 "test/test.byte": pkg_pcre
 "test/test.byte": pkg_str
 "test/test.byte": pkg_threads
@@ -39,7 +39,7 @@ true: annot, bin_annot
 "test/test.byte": use_expect-str
 <test/*.ml{,i,y}>: pkg_batteries
 <test/*.ml{,i,y}>: pkg_bytes
-<test/*.ml{,i,y}>: pkg_oUnit
+<test/*.ml{,i,y}>: pkg_ounit2
 <test/*.ml{,i,y}>: pkg_pcre
 <test/*.ml{,i,y}>: pkg_str
 <test/*.ml{,i,y}>: pkg_threads
Index: ocaml-expect-0.0.6/setup.ml
===================================================================
--- ocaml-expect-0.0.6.orig/setup.ml
+++ ocaml-expect-0.0.6/setup.ml
@@ -7540,7 +7540,7 @@ let setup_t =
                       bs_build_depends =
                         [
                            FindlibPackage
-                             ("oUnit",
+                             ("ounit2",
                                Some (OASISVersion.VGreaterEqual "2.0.0"));
                            InternalLibrary "expect";
                            InternalLibrary "expect-str";
