Index: ocaml-cohttp-2.4.0/cohttp-lwt-jsoo/src/cohttp_lwt_xhr.ml
===================================================================
--- ocaml-cohttp-2.4.0.orig/cohttp-lwt-jsoo/src/cohttp_lwt_xhr.ml
+++ ocaml-cohttp-2.4.0/cohttp-lwt-jsoo/src/cohttp_lwt_xhr.ml
@@ -200,11 +200,11 @@ module Make_client_async(P : Params) = M
                           (File.CoerceTo.arrayBuffer xml##.response)
                           (fun () -> Firebug.console##log
                              (Js.string "XHR Response is not an arrayBuffer; using responseText");
-                             `String xml##.responseText
+                             `String (Js.Opt.get xml##.responseText (fun () -> Js.string ""))
                           )
                           (fun ab -> `ArrayBuffer ab)
                       else
-                        `String xml##.responseText
+                        `String (Js.Opt.get xml##.responseText (fun () -> Js.string ""))
                     in
                     Bb.get b
                  in
@@ -293,11 +293,11 @@ module Make_client_sync(P : Params) = Ma
            (File.CoerceTo.arrayBuffer xml##.response)
            (fun () -> Firebug.console##log
               (Js.string "XHR Response is not an arrayBuffer; using responseText");
-              `String xml##.responseText
+              `String (Js.Opt.get xml##.responseText (fun () -> Js.string ""))
                           )
            (fun ab -> `ArrayBuffer ab)
        else
-         `String xml##.responseText
+         `String (Js.Opt.get xml##.responseText (fun () -> Js.string ""))
      in
      Bb.get b
   in
