Index: mdx-2.4.1/lib/top/compat_top.ml
===================================================================
--- mdx-2.4.1.orig/lib/top/compat_top.ml
+++ mdx-2.4.1/lib/top/compat_top.ml
@@ -97,8 +97,12 @@ let extension_constructor
   ; ext_loc
   ; ext_attributes
 #if OCAML_VERSION >= (4, 11, 0)
+#if OCAML_VERSION >= (5, 3, 0)
+  ; ext_uid = Uid.mk ~current_unit:None
+#else
   ; ext_uid = Uid.mk ~current_unit:"mdx"
 #endif
+#endif
   }
 
 let match_env
