Index: stdcompat-21.1/tools/compiler_version.ml
===================================================================
--- stdcompat-21.1.orig/tools/compiler_version.ml
+++ stdcompat-21.1/tools/compiler_version.ml
@@ -88,6 +88,7 @@ let v5_2_0 = mk 5 2 0
 let v5_2_1 = mk 5 2 1
 let v5_3_0 = mk 5 3 0
 let v5_4_0 = mk 5 4 0
+let v5_4_1 = mk 5 4 1
 
 let known_versions =
 [
@@ -135,6 +136,7 @@ let known_versions =
   v5_2_1;
   v5_3_0;
   v5_4_0;
+  v5_4_1;
 ]
 
 let is_known v = List.mem v known_versions
Index: stdcompat-21.1/tools/compiler_version.mli
===================================================================
--- stdcompat-21.1.orig/tools/compiler_version.mli
+++ stdcompat-21.1/tools/compiler_version.mli
@@ -52,6 +52,7 @@ val v5_2_0 : t
 val v5_2_1 : t
 val v5_3_0 : t
 val v5_4_0 : t
+val v5_4_1 : t
 
 val known_versions : t list
 
