Index: virtual_dom-0.11.0/src/tyxml.ml
===================================================================
--- virtual_dom-0.11.0.orig/src/tyxml.ml
+++ virtual_dom-0.11.0/src/tyxml.ml
@@ -23,6 +23,7 @@ module Xml = struct
   type event_handler = Dom_html.event Js.t -> Event.t
   type mouse_event_handler = Dom_html.mouseEvent Js.t -> Event.t
   type keyboard_event_handler = Dom_html.keyboardEvent Js.t -> Event.t
+  type touch_event_handler = Dom_html.touchEvent Js.t -> Event.t
   type attrib = Attr.t
 
   let attr name value =
@@ -47,6 +48,8 @@ module Xml = struct
     attr_ev name value
   let keyboard_event_handler_attrib name (value : keyboard_event_handler) =
     attr_ev name value
+  let touch_event_handler_attrib name (value : touch_event_handler) =
+    attr_ev name value
   let uri_attrib name value = attr name value
   let uris_attrib name values = attr name (String.concat " " values)
 
