Index: dose3-5.0.1/common/input.ml
===================================================================
--- dose3-5.0.1.orig/common/input.ml
+++ dose3-5.0.1/common/input.ml
@@ -35,10 +35,10 @@ let bzip_open_file file =
   (* workaround to avoid segfault :
    * http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=602170 *)
   let _ = Bz2.version in 
-  let s = " " in
+  let s = Bytes.make 1 ' ' in
   let ch = Bz2.open_in (open_in file) in
   let input_char ch = 
-    try ignore (Bz2.read ch s 0 1) ; s.[0]
+    try ignore (Bz2.read ch s 0 1) ; Bytes.get s 0
     with End_of_file -> raise IO.No_more_input
   in
   let read ch s pos len =
@@ -47,7 +47,7 @@ let bzip_open_file file =
   in
   IO.create_in
   ~read:(fun () -> input_char ch)
-  ~input:(fun x -> read ch (Bytes.to_string x))
+  ~input:(fun x -> read ch x)
   ~close:(fun () -> Bz2.close_in ch)
 #else
     fatal "bzip not supported. re-configure with --with-bz2"
