Index: dose3-5.0.1/algo/dominators.ml
===================================================================
--- dose3-5.0.1.orig/algo/dominators.ml
+++ dose3-5.0.1/algo/dominators.ml
@@ -30,6 +30,13 @@ include Util.Logging(struct let label =
 module G = Defaultgraphs.PackageGraph.G
 module O = Defaultgraphs.GraphOper(G)
 module S = Defaultgraphs.PackageGraph.S
+module I = struct
+	include G
+
+	let empty () = create ()
+	let add_edge g a b = let () = add_edge g a b in g
+
+end
 
 let impactset (graph,pkg) = G.fold_pred S.add graph pkg (S.singleton pkg)
 let scons (graph,pkg) = G.fold_succ S.add graph pkg (S.singleton pkg)
@@ -102,7 +109,7 @@ let dominators_tarjan graph =
 
   Util.Timer.start tjntimer;
 #if OCAMLGRAPHVERSION >= 186
-  let module Dom = Dominator.Make_graph(G) in
+  let module Dom = Dominator.Make_graph(I) in
 #else
   let module Dom = Dominator.Make(G) in
 #endif
