Index: lablgl-1.06/src/Makefile
===================================================================
--- lablgl-1.06.orig/src/Makefile
+++ lablgl-1.06/src/Makefile
@@ -8,7 +8,7 @@ LIBS = $(GLLIBS) $(XLIBS)
 LIBDIRS =
 
 
-OCAMLINC=-I +labltk
+OCAMLINC=-I +labltk -I +camlp-streams
 
 # Files
 LIBOBJS =	raw.cmo \
@@ -32,10 +32,10 @@ opt: lablgl.cmxa
 tools: var2def$(XE) var2switch$(XE)
 
 var2def$(XE): var2def.cmo
-	$(LINKER) $< -o $@
+	$(OCAMLFIND) $(LINKER) -package camlp-streams -linkpkg $< -o $@
 
 var2switch$(XE): var2switch.cmo
-	$(LINKER) $< -o $@
+	$(OCAMLFIND) $(LINKER) -package camlp-streams -linkpkg $< -o $@
 
 ifeq ($(TOOLCHAIN), msvc)
 liblablgl$(XA): $(COBJS)
