Index: mirage-profile-0.8.1/unix/jbuild
===================================================================
--- mirage-profile-0.8.1.orig/unix/jbuild
+++ mirage-profile-0.8.1/unix/jbuild
@@ -1,7 +1,7 @@
 (library
   ((name mirage_profile_unix)
    (public_name mirage-profile-unix)
-   (libraries (cstruct mtime.clock.os))
+   (libraries (cstruct mtime.clock.os ocplib-endian))
    (wrapped false)
 ))
 
