Index: ppx_string-0.14.0/src/dune
===================================================================
--- ppx_string-0.14.0.orig/src/dune
+++ ppx_string-0.14.0/src/dune
@@ -1,4 +1,4 @@
 (library (name ppx_string) (public_name ppx_string) (kind ppx_rewriter)
- (libraries base compiler-libs.common ppxlib)
+ (libraries base compiler-libs.common ppxlib stdio)
  (preprocess
-  (pps ppx_base -annotated-ignores ppxlib.metaquot ppxlib.traverse)))
\ No newline at end of file
+  (pps ppx_base -annotated-ignores ppxlib.metaquot ppxlib.traverse)))
