Index: ppx_protocol_conv-1.0.0/drivers/json.ml
===================================================================
--- ppx_protocol_conv-1.0.0.orig/drivers/json.ml
+++ ppx_protocol_conv-1.0.0/drivers/json.ml
@@ -58,7 +58,7 @@ let to_record: type a b. ?flags:flag ->
       Yojson.Safe.Util.to_assoc t
       |> List.fold_left
         ~init:(Map.Using_comparator.empty ~comparator:String.comparator)
-        ~f:(fun m (key, data) -> Map.add ~key ~data m)
+        ~f:(fun m (key, data) -> Map.add_exn ~key ~data m)
     in
     f values
 
Index: ppx_protocol_conv-1.0.0/drivers/msgpack.ml
===================================================================
--- ppx_protocol_conv-1.0.0.orig/drivers/msgpack.ml
+++ ppx_protocol_conv-1.0.0/drivers/msgpack.ml
@@ -63,7 +63,7 @@ let to_record: type a b. ?flags:flag ->
         List.fold_left
           ~init:(Map.Using_comparator.empty ~comparator:String.comparator)
           ~f:(fun m ->
-              function | (Msgpck.String key, data) -> Map.add ~key ~data m
+              function | (Msgpck.String key, data) -> Map.add_exn ~key ~data m
                        | (e, _) -> raise_errorf e "string expected")
           m
       | e -> raise_errorf e "map expected"
Index: ppx_protocol_conv-1.0.0/ppx/ppx_protocol_conv.ml
===================================================================
--- ppx_protocol_conv-1.0.0.orig/ppx/ppx_protocol_conv.ml
+++ ppx_protocol_conv-1.0.0/ppx/ppx_protocol_conv.ml
@@ -1,4 +1,3 @@
-open Ppx_type_conv.Std
 open Ppx_core
 open Ast_builder.Default
 open !Printf
