Index: ppx_protocol_conv-5.1.3/ppx/dune
===================================================================
--- ppx_protocol_conv-5.1.3.orig/ppx/dune
+++ ppx_protocol_conv-5.1.3/ppx/dune
@@ -2,7 +2,7 @@
  (name ppx_protocol_conv)
  (public_name ppx_protocol_conv)
  (kind ppx_deriver)
- (libraries base ppxlib)
+ (libraries base ppxlib stdio)
  (preprocess (pps ppxlib.metaquot))
  (synopsis "ppx to derive (de)serializers of a type")
 )
Index: ppx_protocol_conv-5.1.3/ppx_protocol_conv.opam
===================================================================
--- ppx_protocol_conv-5.1.3.orig/ppx_protocol_conv.opam
+++ ppx_protocol_conv-5.1.3/ppx_protocol_conv.opam
@@ -14,6 +14,7 @@ depends: [
   "base" {>= "v0.14.0" }
   "dune" {>= "1.2"}
   "ppxlib" {>= "0.9.0"}
+  "stdio"
   "ppx_sexp_conv" {with-test}
   "sexplib" {with-test}
   "alcotest" {with-test & >= "0.8.0"}
