Index: gettext-v0.4.2/src/lib/gettext-camomile/gettextCamomile.ml
===================================================================
--- gettext-v0.4.2.orig/src/lib/gettext-camomile/gettextCamomile.ml
+++ gettext-v0.4.2/src/lib/gettext-camomile/gettextCamomile.ml
@@ -20,7 +20,7 @@
 (*  USA                                                                   *)
 (**************************************************************************)
 
-open CamomileLibraryDefault.Camomile
+open CamomileLib
 open GettextTypes
 
 (** Error reported when something goes wrong during Camomile initialization.
@@ -39,16 +39,16 @@ module Charset : GettextCharset.CHARSET_
 
   type u = {
     failsafe : failsafe;
-    in_enc : CharEncoding.t;
-    out_enc : CharEncoding.t;
+    in_enc : Camomile.CharEncoding.t;
+    out_enc : Camomile.CharEncoding.t;
   }
 
   let create t in_enc out_enc =
     try
       {
         failsafe = t.GettextTypes.failsafe;
-        in_enc = CharEncoding.of_name in_enc;
-        out_enc = CharEncoding.of_name out_enc;
+        in_enc = Camomile.CharEncoding.of_name in_enc;
+        out_enc = Camomile.CharEncoding.of_name out_enc;
       }
     with e ->
       GettextUtils.fail_or_continue t.GettextTypes.failsafe
@@ -58,18 +58,18 @@ module Charset : GettextCharset.CHARSET_
              e ))
         {
           failsafe = t.GettextTypes.failsafe;
-          in_enc = CharEncoding.of_name "ISO-8859-1";
-          out_enc = CharEncoding.of_name "ISO-8859-1";
+          in_enc = Camomile.CharEncoding.of_name "ISO-8859-1";
+          out_enc = Camomile.CharEncoding.of_name "ISO-8859-1";
         }
 
   let recode u str =
-    try CharEncoding.recode_string ~in_enc:u.in_enc ~out_enc:u.out_enc str
+    try Camomile.CharEncoding.recode_string ~in_enc:u.in_enc ~out_enc:u.out_enc str
     with e ->
       GettextUtils.fail_or_continue u.failsafe
         (GettextCamomileCreate
            ( Printf.sprintf "Cannot create convert string %s from %s to %s" str
-               (CharEncoding.name_of u.in_enc)
-               (CharEncoding.name_of u.out_enc),
+               (Camomile.CharEncoding.name_of u.in_enc)
+               (Camomile.CharEncoding.name_of u.out_enc),
              e ))
         str
 end
