From 7b935051d1ac1e8ff3b5735522a613e89f99a3be Mon Sep 17 00:00:00 2001
From: Gabriel Scherer <gabriel.scherer@gmail.com>
Date: Thu, 11 Jul 2019 14:59:30 +0200
Subject: [PATCH 14/42] re-export Bigarray.*.map_file functions removed in 4.08

---
 src/batBigarray.mlv | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/src/batBigarray.mlv b/src/batBigarray.mlv
index e498164d..aae051b0 100644
--- a/src/batBigarray.mlv
+++ b/src/batBigarray.mlv
@@ -153,6 +153,8 @@ module Genarray =
 struct
   include Bigarray.Genarray
 
+##V>=4.8##let map_file = Unix.map_file
+
   let ofs e = ofs_of_layout (layout e)
 
 ##V<4.3##  let size_in_bytes arr =
@@ -314,6 +316,10 @@ let array1_of_genarray = Bigarray.array1_of_genarray
 module Array1 = struct
   include Bigarray.Array1
 
+##V>=4.8##let map_file fd ?pos kind layout shared dim =
+##V>=4.8##  Bigarray.array1_of_genarray
+##V>=4.8##    (Unix.map_file fd ?pos kind layout shared [|dim|])
+
   let ofs e = ofs_of_layout (layout e)
 
 ##V<4.3##  let size_in_bytes arr =
@@ -371,6 +377,10 @@ end
 module Array2 = struct
   include Bigarray.Array2
 
+##V>=4.8##let map_file fd ?pos kind layout shared dim1 dim2 =
+##V>=4.8##  Bigarray.array2_of_genarray
+##V>=4.8##    (Unix.map_file fd ?pos kind layout shared [|dim1; dim2|])
+
   let ofs e = ofs_of_layout (layout e)
 
 ##V<4.3##  let size_in_bytes arr =
@@ -425,6 +435,10 @@ end
 module Array3 = struct
   include Bigarray.Array3
 
+##V>=4.8##let map_file fd ?pos kind layout shared dim1 dim2 dim3 =
+##V>=4.8##  Bigarray.array3_of_genarray
+##V>=4.8##    (Unix.map_file fd ?pos kind layout shared [|dim1; dim2; dim3|])
+
   let ofs e = ofs_of_layout (layout e)
 
 ##V<4.3##  let size_in_bytes arr =
-- 
2.22.0

