commit 5d67375f4c81b06238309965764353cf03cdd099
Author: Gabriel Scherer <gabriel.scherer@gmail.com>
Date:   Tue Apr 19 20:21:07 2016 -0400

    safe-string: make Base64 safe

diff --git a/src/batBase64.ml b/src/batBase64.ml
index 17e0e54b..4660f9eb 100644
--- a/src/batBase64.ml
+++ b/src/batBase64.ml
@@ -67,7 +67,7 @@ let encode ?(tbl=chars) ch =
   in
   let output s p l =
     for i = p to p + l - 1 do
-      write (String.unsafe_get s i)
+      write (Bytes.unsafe_get s i)
     done;
     l
   in
