Index: batteries-included-3.5.1/src/dune
===================================================================
--- batteries-included-3.5.1.orig/src/dune
+++ batteries-included-3.5.1/src/dune
@@ -5,7 +5,7 @@
  (preprocess
   (action (run build/prefilter.exe %{input-file})))
  (flags (:standard -w -3-32-52))
- (libraries num threads)
+ (libraries num threads camlp-streams)
  (inline_tests
    (backend qtest_batteries)
    (deps %{project_root}/qtest/qtest_preamble.ml)
@@ -68,8 +68,12 @@ batPervasives batSet batSplay batTuple b
 (rule (copy batBytesCompat.mlv batBytesCompat.ml))
 (rule (copy batChar.mlv batChar.ml))
 (rule (copy batDigest.mlv batDigest.ml))
+(rule (copy batEither.mlv batEither.ml))
+(rule (copy batEither.mliv batEither.mli))
 (rule (copy batFilename.mlv batFilename.ml))
 (rule (copy batFilename.mliv batFilename.mli))
+(rule (copy batFloat.mliv batFloat.mli))
+(rule (copy batFloat.mlv batFloat.ml))
 (rule (copy batFormat.mlv batFormat.ml))
 (rule (copy batFormat.mliv batFormat.mli))
 (rule (copy batGc.mliv batGc.mli))
@@ -87,6 +91,8 @@ batPervasives batSet batSplay batTuple b
 (rule (copy batList.mliv batList.mli))
 (rule (copy batMarshal.mlv batMarshal.ml))
 (rule (copy batMarshal.mliv batMarshal.mli))
+(rule (copy batMap.mliv batMap.mli))
+(rule (copy batMap.mlv batMap.ml))
 (rule (copy batNativeint.mlv batNativeint.ml))
 (rule (copy batNativeint.mliv batNativeint.mli))
 (rule (copy batPervasives.mliv batPervasives.mli))
