commit 8cbb5d5809b754841d6d278dea777e865404f724
Author: Gabriel Scherer <gabriel.scherer@gmail.com>
Date:   Sun Apr 17 13:53:49 2016 -0400

    safe-string: make Genlex safe

diff --git a/src/batGenlex.ml b/src/batGenlex.ml
index be772401..7fc720a0 100644
--- a/src/batGenlex.ml
+++ b/src/batGenlex.ml
@@ -51,16 +51,16 @@ let to_enum_filter kwd_table =
   let reset_buffer () = buffer := initial_buffer; bufpos := 0 in
 
   let store c =
-    if !bufpos >= String.length !buffer then
+    if !bufpos >= Bytes.length !buffer then
       begin
         let newbuffer = Bytes.create (2 * !bufpos) in
-        String.blit !buffer 0 newbuffer 0 !bufpos; buffer := newbuffer
+        Bytes.blit !buffer 0 newbuffer 0 !bufpos; buffer := newbuffer
       end;
     Bytes.set !buffer !bufpos c;
     incr bufpos in
 
   let get_string () =
-    let s = String.sub !buffer 0 !bufpos in buffer := initial_buffer; s
+    let s = Bytes.sub_string !buffer 0 !bufpos in buffer := initial_buffer; s
   in
   let ident_or_keyword id =
     try Hashtbl.find kwd_table id with
