commit 8327cf0c8a6d9539d03b8dfaad9e478b19435a51
Author: Gabriel Scherer <gabriel.scherer@gmail.com>
Date:   Sun Apr 17 13:51:35 2016 -0400

    safe-string: make Lexing safe

diff --git a/src/batLexing.mli b/src/batLexing.mli
index 9aec1612..78a2b94d 100644
--- a/src/batLexing.mli
+++ b/src/batLexing.mli
@@ -60,7 +60,7 @@ val dummy_pos : position;;
 
 type lexbuf = Lexing.lexbuf =
   { refill_buff : lexbuf -> unit;
-    mutable lex_buffer : string;
+    mutable lex_buffer : bytes;
     mutable lex_buffer_len : int;
     mutable lex_abs_pos : int;
     mutable lex_start_pos : int;
@@ -98,12 +98,12 @@ val from_string : string -> lexbuf
     the string. An end-of-input condition is generated when the
     end of the string is reached. *)
 
-val from_function : (string -> int -> int) -> lexbuf
+val from_function : (Bytes.t -> int -> int) -> lexbuf
 (** Create a lexer buffer with the given function as its reading method.
     When the scanner needs more characters, it will call the given
-    function, giving it a character string [s] and a character
-    count [n]. The function should put [n] characters or less in [s],
-    starting at character number 0, and return the number of characters
+    function, giving it a byte sequence [s] and a byte
+    count [n]. The function should put [n] bytes or less in [s],
+    starting at byte number 0, and return the number of byte
     provided. A return value of 0 means end of input. *)
 
 
