commit 1c5f96a702ed329b0c4178cc94145ef8f9e79fff
Author: Gabriel Scherer <gabriel.scherer@gmail.com>
Date:   Tue Apr 19 17:25:39 2016 -0400

    safe-string: make Printf safe

diff --git a/src/batPrintf.mlv b/src/batPrintf.mlv
index 26831792..2d70fd82 100644
--- a/src/batPrintf.mlv
+++ b/src/batPrintf.mlv
@@ -97,11 +97,11 @@ let parse_string_conversion sfmt =
 let pad_string pad_char p neg s i len =
   if p = len && i = 0 then s else
   if p <= len then String.sub s i len else
-    let res = String.make p pad_char in
+    let res = Bytes.make p pad_char in
     if neg
-    then String.blit s i res 0 len
-    else String.blit s i res (p - len) len;
-    res
+    then Bytes.blit_string s i res 0 len
+    else Bytes.blit_string s i res (p - len) len;
+    Bytes.unsafe_to_string res
 
 (* Format a string given a %s format, e.g. %40s or %-20s.
    To do: ignore other flags (#, +, etc)? *)
@@ -134,8 +134,9 @@ let extract_format_int conv fmt start stop widths =
   let sfmt = extract_format fmt start stop widths in
   match conv with
   | 'n' | 'N' ->
-    Bytes.set sfmt (String.length sfmt - 1) 'u';
-    sfmt
+    let sfmt = Bytes.of_string sfmt in
+    Bytes.set sfmt (Bytes.length sfmt - 1) 'u';
+    Bytes.unsafe_to_string sfmt
   | _ -> sfmt;;
 
 (* Returns the position of the next character following the meta format
