commit bcb538ca84209c2d3ea251a455096b30dfc35627
Author: Gabriel Scherer <gabriel.scherer@gmail.com>
Date:   Tue Apr 19 20:20:19 2016 -0400

    safe-string: make Unix safe

diff --git a/src/batUnix.mlv b/src/batUnix.mlv
index e122e16f..9b09a289 100644
--- a/src/batUnix.mlv
+++ b/src/batUnix.mlv
@@ -38,7 +38,7 @@ let run_and_read cmd =
     begin
       let was_read = ref (input ic line_buff 0 buff_size) in
       while !was_read <> 0 do
-        Buffer.add_substring buff line_buff 0 !was_read;
+        Buffer.add_subbytes buff line_buff 0 !was_read;
         was_read := input ic line_buff 0 buff_size;
       done;
       close_in ic;
